{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.10.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"nvidiaTeslaT4","dataSources":[{"sourceId":87793,"databundleVersionId":12276181,"sourceType":"competition"},{"sourceId":10855324,"sourceType":"datasetVersion","datasetId":6742586},{"sourceId":11118830,"sourceType":"datasetVersion","datasetId":6933267},{"sourceId":224830487,"sourceType":"kernelVersion"}],"dockerImageVersionId":30919,"isInternetEnabled":false,"language":"python","sourceType":"notebook","isGpuEnabled":true}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"MODEL_TYPE='protenix'\nVALIDATION=False","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T05:30:35.236153Z","iopub.execute_input":"2025-05-14T05:30:35.236451Z","iopub.status.idle":"2025-05-14T05:30:35.239908Z","shell.execute_reply.started":"2025-05-14T05:30:35.236427Z","shell.execute_reply":"2025-05-14T05:30:35.239075Z"}},"outputs":[],"execution_count":13},{"cell_type":"code","source":"!pip show biopython\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T05:29:54.556683Z","iopub.execute_input":"2025-05-14T05:29:54.557050Z","iopub.status.idle":"2025-05-14T05:29:57.702871Z","shell.execute_reply.started":"2025-05-14T05:29:54.556993Z","shell.execute_reply":"2025-05-14T05:29:57.701954Z"}},"outputs":[{"name":"stdout","text":"Name: biopython\nVersion: 1.85\nSummary: Freely available tools for computational molecular biology.\nHome-page: https://biopython.org/\nAuthor: The Biopython Contributors\nAuthor-email: biopython@biopython.org\nLicense: \nLocation: /usr/local/lib/python3.10/dist-packages\nRequires: numpy\nRequired-by: protenix\n","output_type":"stream"}],"execution_count":8},{"cell_type":"markdown","source":"## Install requirements ","metadata":{}},{"cell_type":"markdown","source":"RNA 3D 예측 결과와 실제 구조를 비교하고, 이를 평가하는 파이프 라인.","metadata":{}},{"cell_type":"code","source":"if MODEL_TYPE=='protenix' and VALIDATION:\n    !pip install --no-deps protenix\n    !pip install biopython\n    !pip install ml-collections\n    !pip install biotite==1.0.1\n    !pip install rdkit\n!export PROTENIX_DATA_ROOT_DIR=/kaggle/input/protenix-checkpoints","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T05:30:36.789141Z","iopub.execute_input":"2025-05-14T05:30:36.789407Z","iopub.status.idle":"2025-05-14T05:30:36.908506Z","shell.execute_reply.started":"2025-05-14T05:30:36.789386Z","shell.execute_reply":"2025-05-14T05:30:36.907431Z"}},"outputs":[],"execution_count":14},{"cell_type":"code","source":"! mkdir /af3-dev \n! ln -s /kaggle/input/protenix-checkpoints /af3-dev/release_data\n! ls /af3-dev/release_data/","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:14.080698Z","iopub.execute_input":"2025-05-14T02:15:14.081025Z","iopub.status.idle":"2025-05-14T02:15:14.436581Z","shell.execute_reply.started":"2025-05-14T02:15:14.081002Z","shell.execute_reply":"2025-05-14T02:15:14.435572Z"}},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Helper scripts","metadata":{}},{"cell_type":"code","source":"import Bio\n\nfrom copy import deepcopy\n\nimport pandas as pd\nfrom Bio.PDB import Atom, Model, Chain, Residue, Structure, PDBParser\nfrom Bio import SeqIO\nimport os, sys\nimport re\nimport numpy as np\nimport torch\n\nimport matplotlib\nimport matplotlib.pyplot as plt\nfrom tqdm import tqdm\nimport time\ntime0=time.time()\nfrom Bio import AlignIO\nprint('IMPORT OK !!!!')","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:17.453248Z","iopub.execute_input":"2025-05-14T02:15:17.453637Z","iopub.status.idle":"2025-05-14T02:15:19.510848Z","shell.execute_reply.started":"2025-05-14T02:15:17.453606Z","shell.execute_reply":"2025-05-14T02:15:19.510032Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"PYTHON = sys.executable\nprint('PYTHON',PYTHON)\n\nRHONET_DIR=\\\n'/kaggle/input/data-for-demo-for-rhofold-plus-with-kaggle-msa/RhoFold-main'\n#'<your downloaded rhofold repo>/RhoFold-main'\n\nUSALIGN = \\\n'/kaggle/working//USalign'\n#'<your us align path>/USalign'\n\nos.system('cp /kaggle/input/usalign/USalign /kaggle/working/')\nos.system('sudo chmod u+x /kaggle/working//USalign')\nsys.path.append(RHONET_DIR)\n\n\nDATA_KAGGLE_DIR = '/kaggle/input/stanford-rna-3d-folding'\n\n\n# helper ----\n# 딕셔너리 <-> 키 매핑\nclass dotdict(dict):\n\t__setattr__ = dict.__setitem__\n\t__delattr__ = dict.__delitem__\n\n\tdef __getattr__(self, name):\n\t\ttry:\n\t\t\treturn self[name]\n\t\texcept KeyError:\n\t\t\traise AttributeError(name)\n\n# visualisation helper ----\ndef set_aspect_equal(ax):\n\tx_limits = ax.get_xlim()\n\ty_limits = ax.get_ylim()\n\tz_limits = ax.get_zlim()\n\n\t# Compute the mean of each axis\n\tx_middle = np.mean(x_limits)\n\ty_middle = np.mean(y_limits)\n\tz_middle = np.mean(z_limits)\n\n\t# Compute the max range across all axes\n\tmax_range = max(x_limits[1] - x_limits[0],\n\t\t\t\t\ty_limits[1] - y_limits[0],\n\t\t\t\t\tz_limits[1] - z_limits[0]) / 2.0\n\n\t# Set the new limits to ensure equal scaling\n\tax.set_xlim(x_middle - max_range, x_middle + max_range)\n\tax.set_ylim(y_middle - max_range, y_middle + max_range)\n\tax.set_zlim(z_middle - max_range, z_middle + max_range)\n\n\n\n\n# xyz df helper --------------------\ndef get_truth_df(target_id):\n    truth_df = LABEL_DF[LABEL_DF['target_id'] == target_id]\n    truth_df = truth_df.reset_index(drop=True)\n    return truth_df\n\ndef parse_output_to_df(output, seq, target_id):\n    df = []\n    chain_data = []\n    for i, res in enumerate(seq):\n        d=dict(ID = target_id,\n                    resname=res,\n                    resid=i+1)\n        for n in range(len(output)):\n            d={**d, f'x_{n+1}': round(output[n,i,0].item(),3),\n                     f'y_{n+1}': round(output[n,i,1].item(),3),\n                     f'z_{n+1}': round(output[n,i,2].item(),3)}\n        chain_data.append(d)\n\n    if len(chain_data)!=0:\n        chain_df = pd.DataFrame(chain_data)\n        df.append(chain_df)\n        ##print(chain_df)\n    return df\n\ndef parse_pdb_to_df(pdb_file, target_id):\n    parser = PDBParser()\n    structure = parser.get_structure('', pdb_file)\n\n    df = []\n    for model in structure:\n        for chain in model:\n            print(chain)\n            chain_data = []\n            for residue in chain:\n                # print(residue)\n                if residue.get_resname() in ['A', 'U', 'G', 'C']:\n                    # Check if the residue has a C1' atom\n                    if 'C1\\'' in residue:\n                        atom = residue['C1\\'']\n                        xyz = atom.get_coord()\n                        resname = residue.get_resname()\n                        resid = residue.get_id()[1]\n\n                        #todo detect discontinous: resid = prev_resid+1\n                        #ID\tresname\tresid\tx_1\ty_1\tz_1\n                        chain_data.append(dict(\n                            ID = target_id+'_'+str(resid),\n                            resname=resname,\n                            resid=resid,\n                            x_1=xyz[0],\n                            y_1=xyz[1],\n                            z_1=xyz[2],\n                        ))\n                        ##print(f\"Residue {resname} {resid}, Atom: {atom.get_name()}, xyz: {xyz}\")\n\n            if len(chain_data)!=0:\n                chain_df = pd.DataFrame(chain_data)\n                df.append(chain_df)\n                ##print(chain_df)\n    return df\n\n# usalign helper --------------------\ndef write_target_line(\n    atom_name, atom_serial, residue_name, chain_id, residue_num, x_coord, y_coord, z_coord, occupancy=1.0, b_factor=0.0, atom_type='P'\n):\n    \"\"\"\n    Writes a single line of PDB format based on provided atom information.\n\n    Args:\n        atom_name (str): Name of the atom (e.g., \"N\", \"CA\").\n        atom_serial (int): Atom serial number.\n        residue_name (str): Residue name (e.g., \"ALA\").\n        chain_id (str): Chain identifier.\n        residue_num (int): Residue number.\n        x_coord (float): X coordinate.\n        y_coord (float): Y coordinate.\n        z_coord (float): Z coordinate.\n        occupancy (float, optional): Occupancy value (default: 1.0).\n        b_factor (float, optional): B-factor value (default: 0.0).\n\n    Returns:\n        str: A single line of PDB string.\n    \"\"\"\n    return f'ATOM  {atom_serial:>5d}  {atom_name:<5s} {residue_name:<3s} {residue_num:>3d}    {x_coord:>8.3f}{y_coord:>8.3f}{z_coord:>8.3f}{occupancy:>6.2f}{b_factor:>6.2f}           {atom_type}\\n'\n\ndef write_xyz_to_pdb(df, pdb_file, xyz_id = 1):\n    resolved_cnt = 0\n    with open(pdb_file, 'w') as target_file:\n        for _, row in df.iterrows():\n            x_coord = row[f'x_{xyz_id}']\n            y_coord = row[f'y_{xyz_id}']\n            z_coord = row[f'z_{xyz_id}']\n\n            if x_coord > -1e17 and y_coord > -1e17 and z_coord > -1e17:\n                resolved_cnt += 1\n                target_line = write_target_line(\n                    atom_name=\"C1'\",\n                    atom_serial=int(row['resid']),\n                    residue_name=row['resname'],\n                    chain_id='0',\n                    residue_num=int(row['resid']),\n                    x_coord=x_coord,\n                    y_coord=y_coord,\n                    z_coord=z_coord,\n                    atom_type='C',\n                )\n                target_file.write(target_line)\n    return resolved_cnt\n\ndef parse_usalign_for_tm_score(output):\n    # Extract TM-score based on length of reference structure (second)\n    tm_score_match = re.findall(r'TM-score=\\s+([\\d.]+)', output)[1]\n    if not tm_score_match:\n        raise ValueError('No TM score found')\n    return float(tm_score_match)\n\ndef parse_usalign_for_transform(output):\n    # Locate the rotation matrix section\n    matrix_lines = []\n    found_matrix = False\n\n    for line in output.splitlines():\n        if \"The rotation matrix to rotate Structure_1 to Structure_2\" in line:\n            found_matrix = True\n        elif found_matrix and re.match(r'^\\d+\\s+[-\\d.]+\\s+[-\\d.]+\\s+[-\\d.]+\\s+[-\\d.]+$', line):\n            matrix_lines.append(line)\n        elif found_matrix and not line.strip():\n            break  # Stop parsing if an empty line is encountered after the matrix\n\n    # Parse the rotation matrix values\n    rotation_matrix = []\n    for line in matrix_lines:\n        parts = line.split()\n        row_values = list(map(float, parts[1:]))  # Skip the first column (index)\n        rotation_matrix.append(row_values)\n\n    return np.array(rotation_matrix)\n\ndef call_usalign(predict_df, truth_df, verbose=1):\n    truth_pdb = '~truth.pdb'\n    predict_pdb = '~predict.pdb'\n    write_xyz_to_pdb(predict_df, predict_pdb, xyz_id=1)\n    write_xyz_to_pdb(truth_df, truth_pdb, xyz_id=1)\n\n    command = f'{USALIGN} {predict_pdb} {truth_pdb} -atom \" C1\\'\" -m -'\n    output = os.popen(command).read()\n    if verbose==1:\n        print(output)\n    tm_score = parse_usalign_for_tm_score(output)\n    transform = parse_usalign_for_transform(output)\n    return tm_score, transform\n\nprint('HELPER OK!!!')","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:18:45.504501Z","iopub.execute_input":"2025-05-14T02:18:45.505006Z","iopub.status.idle":"2025-05-14T02:18:45.630154Z","shell.execute_reply.started":"2025-05-14T02:18:45.504964Z","shell.execute_reply":"2025-05-14T02:18:45.629389Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if MODEL_TYPE=='protenix':\n    \n    \n    from runner.batch_inference import get_default_runner\n    from runner.inference import update_inference_configs, InferenceRunner\n\n    from protenix.data.infer_data_pipeline import InferenceDataset\n\n    np.random.seed(0)\n    torch.random.manual_seed(0)\n    torch.cuda.manual_seed_all(0)\n\nclass DictDataset(InferenceDataset):\n    def __init__(\n        self,\n        seq_list: list,\n        dump_dir: str,\n        id_list: list = None,\n        msa_root_1: str = \"/kaggle/input/stanford-rna-3d-folding/MSA\",\n        msa_root_2: str = \"/kaggle/input/stanford-rna-3d-folding/MSA_v2\",\n    ) -> None:\n        self.dump_dir = dump_dir\n        self.msa_roots = [msa_root_2, msa_root_1]  # 우선순위: v2 > v1\n        self.use_msa = True  # Protenix 내부에서 참조하므로 반드시 필요\n\n        if isinstance(id_list, type(None)):\n            self.inputs = [{\"sequences\": [{\"rnaSequence\": {\"sequence\": seq, \"count\": 1}}], \"name\": \"query\"}\n                           for seq in seq_list]\n        else:\n            self.inputs = [{\"sequences\": [{\"rnaSequence\": {\"sequence\": seq, \"count\": 1}}], \"name\": i}\n                           for i, seq in zip(id_list, seq_list)]\n\n    def __getitem__(self, index):\n        data, atom_array, error_msg = super().__getitem__(index)\n        target_id = self.inputs[index][\"name\"]\n\n        msa_tensor = None\n        for root in self.msa_roots:\n            msa_path = os.path.join(root, f\"{target_id}.MSA.fasta\")\n            if os.path.exists(msa_path):\n                try:\n                    msa_tensor = self.load_msa_fasta(msa_path)\n                    print(f\"✅ Loaded MSA for {target_id} from {root} | shape={msa_tensor.shape}\")\n                    break\n                except Exception as e:\n                    print(f\"⚠️ Error loading MSA for {target_id}: {e}\")\n\n        if msa_tensor is not None and msa_tensor.shape[0] >= 10:\n            if \"input_feature_dict\" not in data:\n                data[\"input_feature_dict\"] = {}\n            data[\"input_feature_dict\"][\"msa_feat\"] = msa_tensor\n            print(f\"✅ Injected msa_feat for {target_id} | shape={msa_tensor.shape}\")\n        else:\n            print(f\"⚠️ Skipped msa_feat for {target_id} | depth insufficient or not found\")\n\n        return data, atom_array, error_msg\n\n    def load_msa_fasta(self, msa_path):\n        from Bio import AlignIO\n        alignment = AlignIO.read(msa_path, \"fasta\")\n        alignment = alignment[:64]  # optional: top 64 rows only\n        L = len(alignment[0])\n        N = len(alignment)\n\n        # RNA base vocabulary mapping\n        alphabet = {'A': 0, 'C': 1, 'G': 2, 'U': 3, '-': 4}\n        msa_tensor = torch.zeros((N, L), dtype=torch.long)\n        for i, record in enumerate(alignment):\n            msa_tensor[i] = torch.tensor([\n                alphabet.get(residue.upper(), 4)  # default to gap if unknown\n                for residue in record.seq\n            ])\n        return msa_tensor\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:46:32.380799Z","iopub.execute_input":"2025-05-14T02:46:32.381092Z","iopub.status.idle":"2025-05-14T02:46:32.392705Z","shell.execute_reply.started":"2025-05-14T02:46:32.381069Z","shell.execute_reply":"2025-05-14T02:46:32.392002Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if MODEL_TYPE=='protenix':\n\n    from configs.configs_base import configs as configs_base\n    from configs.configs_data import data_configs\n    from configs.configs_inference import inference_configs\n    from protenix.config.config import parse_configs\n\n    configs_base[\"use_deepspeed_evo_attention\"] = (\n    os.environ.get(\"USE_DEEPSPEED_EVO_ATTENTION\", False) == \"true\")\n    configs_base[\"model\"][\"N_cycle\"] = 10 #10\n    configs_base[\"sample_diffusion\"][\"N_sample\"] = (1 if VALIDATION else 5)\n    configs_base[\"sample_diffusion\"][\"N_step\"] = 200\n    inference_configs['load_checkpoint_path']='/kaggle/input/protenix-checkpoints/model_v0.2.0.pt'\n    configs = {**configs_base, **{\"data\": data_configs}, **inference_configs}\n\n    configs = parse_configs(\n            configs=configs,\n            fill_required_with_null=True,\n        )\n    \n    runner=InferenceRunner(configs)\n    # print(type(runner.model))          # 클래스 이름\n    # print(runner.model)                # 전체 구조 요약","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:29.527953Z","iopub.execute_input":"2025-05-14T02:15:29.528521Z","iopub.status.idle":"2025-05-14T02:15:52.294169Z","shell.execute_reply.started":"2025-05-14T02:15:29.528492Z","shell.execute_reply":"2025-05-14T02:15:52.293241Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if VALIDATION:\n    LABEL_DF = pd.read_csv('/kaggle/input/stanford-rna-3d-folding/train_labels.csv')\n    LABEL_DF['target_id'] = LABEL_DF['ID'].apply(lambda x: '_'.join(x.split('_')[:-1]))\n    train_df=pd.read_csv('/kaggle/input/stanford-rna-3d-folding/train_sequences.csv')\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:18:10.859035Z","iopub.execute_input":"2025-05-14T02:18:10.859347Z","iopub.status.idle":"2025-05-14T02:18:11.085375Z","shell.execute_reply.started":"2025-05-14T02:18:10.859321Z","shell.execute_reply":"2025-05-14T02:18:11.084655Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if MODEL_TYPE == 'protenix' and VALIDATION:\n    import warnings\n    warnings.filterwarnings(\"ignore\")  \n\n    train_df['protenix_tm_score'] = None\n    dataset = DictDataset(\n        seq_list=train_df.sequence,\n        dump_dir='output',\n        id_list=train_df.target_id,\n        msa_root_1=\"/kaggle/input/stanford-rna-3d-folding/MSA\",\n        msa_root_2=\"/kaggle/input/stanford-rna-3d-folding/MSA_v2\",\n    )\n    num_data = len(dataset)\n\n    for i, seq in tqdm(enumerate(train_df.sequence), total=num_data):\n        target_id = train_df.loc[i, 'target_id']\n        print(f\"\\n▶ {i}: {target_id} | len={len(seq)} | TM={train_df.loc[i, 'protenix_tm_score']}\")\n\n        if not pd.isna(train_df.loc[i, 'protenix_tm_score']):\n            print(\"⏭️ TM-score already exists → skip\")\n            continue\n        if len(seq) > 300:\n            print(\"⏭️ Sequence too long → skip\")\n            continue\n\n        truth_df = get_truth_df(target_id)\n        num_valid_coords = sum(~np.isnan(truth_df.x_1))\n        if num_valid_coords < 3:\n            print(\"⏭️ Not enough ground-truth coordinates → skip\")\n            continue\n\n        data, atom_array, data_error_message = dataset[i]\n        if data_error_message != '':\n            print(f\"⏭️ Error loading data: {data_error_message}\")\n            continue\n\n        print(\"📦 input keys:\", data['input_feature_dict'].keys())\n        if \"msa_feat\" in data[\"input_feature_dict\"]:\n            msa_feat = data[\"input_feature_dict\"][\"msa_feat\"]\n            print(\"✅ msa_feat shape:\", msa_feat.shape)\n        else:\n            print(\"❌ msa_feat not present\")\n\n        new_configs = update_inference_configs(configs, data[\"N_token\"].item())\n        runner.update_model_configs(new_configs)\n\n        prediction = runner.predict(data)\n        prediction = prediction['coordinate'][:, data['input_feature_dict']['atom_to_tokatom_idx'] == 12]\n\n        print(f\"🧪 Prediction shape: {prediction.shape}, contains NaN: {torch.isnan(prediction).any().item()}\")\n\n        result = parse_output_to_df(prediction[:1], seq, target_id)[0]\n\n        try:\n            tm_score, transform = call_usalign(result, truth_df, verbose=0)\n            train_df.loc[i, 'protenix_tm_score'] = tm_score\n            print(f\"✅ TM-score: {tm_score:.4f}\")\n        except Exception as e:\n            print(f\"❌ TM-score 계산 실패 (target_id={target_id}): {e}\")\n\n        if (time.time() - time0) > (12 * 3600 - 360):\n            print(\"⏹️ 시간 제한 도달 → 종료\")\n            break\n\n    train_df.to_csv('tm_scores.csv', index=False)\n    print(\"📊 평균 TM-score:\", train_df.protenix_tm_score.mean())\n    display(train_df.protenix_tm_score.hist())\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:46:39.039405Z","iopub.execute_input":"2025-05-14T02:46:39.039718Z"}},"outputs":[],"execution_count":null},{"cell_type":"code","source":"from scipy.ndimage import gaussian_filter1d\n\ndef smooth_coords(df, sigma=1.0):\n    df = df.copy()\n    for n in range(1, 6):  # 5개의 구조 (x_1 ~ x_5, y_1 ~ y_5, z_1 ~ z_5)\n        for axis in ['x', 'y', 'z']:\n            col = f\"{axis}_{n}\"\n            if col in df.columns:\n                df[col] = gaussian_filter1d(df[col].values, sigma=sigma)\n    return df","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if MODEL_TYPE=='protenix' and not VALIDATION:\n    test_df=pd.read_csv('/kaggle/input/stanford-rna-3d-folding/test_sequences.csv')\n    import warnings\n    warnings.filterwarnings(\"ignore\")  \n    \n    dataset = DictDataset(\n        seq_list=test_df.sequence,\n        dump_dir='output',\n        id_list=test_df.target_id,\n        msa_root_1=\"/kaggle/input/stanford-rna-3d-folding/MSA\",\n        msa_root_2=\"/kaggle/input/stanford-rna-3d-folding/MSA_v2\",\n    )   \n    num_data = len(dataset)\n    for i, seq in tqdm(enumerate(test_df.sequence),total=num_data):\n        try:\n            data, atom_array, data_error_message=dataset[i]\n            target_id = data[\"sample_name\"]\n            assert target_id == test_df.target_id[i]\n            assert data_error_message==''\n            \n            new_configs = update_inference_configs(configs, data[\"N_token\"].item())\n            runner.update_model_configs(new_configs)\n            prediction = runner.predict(data)\n            prediction=prediction['coordinate'][:,data['input_feature_dict']['atom_to_tokatom_idx']==12]\n\n            result = parse_output_to_df(prediction, seq, target_id)[0]\n        except:\n            target_id=test_df.target_id[i]\n            print('Failed to predict', target_id)\n            result=pd.DataFrame(columns=['ID', 'resname', 'resid', \n                                         'x_1', 'y_1', 'z_1', \n                                         'x_2', 'y_2', 'z_2',\n                                         'x_3', 'y_3', 'z_3', \n                                         'x_4', 'y_4', 'z_4', \n                                         'x_5', 'y_5', 'z_5'], \n                                         data=[[target_id, x, j+1] + [0.0]*15 for j, x in enumerate(seq)])\n        result['ID']=result.apply(lambda x: x.ID + '_' + str(x.resid), axis=1)\n        result.to_csv('submission.csv', index=False, mode='a', header=(i==0))\n        torch.cuda.empty_cache()\n\n    display(pd.read_csv('submission.csv'))","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"","metadata":{}},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}