{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.10.12","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"gpu","dataSources":[{"sourceId":87793,"databundleVersionId":12276181,"sourceType":"competition"},{"sourceId":10855324,"sourceType":"datasetVersion","datasetId":6742586},{"sourceId":11118830,"sourceType":"datasetVersion","datasetId":6933267},{"sourceId":224830487,"sourceType":"kernelVersion"}],"dockerImageVersionId":30919,"isInternetEnabled":false,"language":"python","sourceType":"notebook","isGpuEnabled":true}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"MODEL_TYPE='protenix'\nVALIDATION=False","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:11:13.785230Z","iopub.execute_input":"2025-05-14T02:11:13.785622Z","iopub.status.idle":"2025-05-14T02:11:13.789496Z","shell.execute_reply.started":"2025-05-14T02:11:13.785587Z","shell.execute_reply":"2025-05-14T02:11:13.788554Z"}},"outputs":[],"execution_count":3},{"cell_type":"code","source":"!pip show biopython\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T00:48:02.397491Z","iopub.execute_input":"2025-05-14T00:48:02.397777Z","iopub.status.idle":"2025-05-14T00:48:03.134862Z","shell.execute_reply.started":"2025-05-14T00:48:02.397746Z","shell.execute_reply":"2025-05-14T00:48:03.134009Z"}},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"## Install requirements ","metadata":{}},{"cell_type":"markdown","source":"RNA 3D 예측 결과와 실제 구조를 비교하고, 이를 평가하는 파이프 라인.","metadata":{}},{"cell_type":"code","source":"if MODEL_TYPE=='protenix' and VALIDATION:\n    !pip install --no-deps protenix\n    !pip install biopython\n    !pip install ml-collections\n    !pip install biotite==1.0.1\n    !pip install rdkit\n!export PROTENIX_DATA_ROOT_DIR=/kaggle/input/protenix-checkpoints","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:11:15.372299Z","iopub.execute_input":"2025-05-14T02:11:15.372621Z","iopub.status.idle":"2025-05-14T02:11:37.189735Z","shell.execute_reply.started":"2025-05-14T02:11:15.372579Z","shell.execute_reply":"2025-05-14T02:11:37.188827Z"}},"outputs":[{"name":"stdout","text":"Collecting protenix\n  Downloading protenix-0.4.6-py3-none-any.whl.metadata (1.1 kB)\nDownloading protenix-0.4.6-py3-none-any.whl (441 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m441.1/441.1 kB\u001b[0m \u001b[31m22.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hInstalling collected packages: protenix\nSuccessfully installed protenix-0.4.6\nCollecting biopython\n  Downloading biopython-1.85-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\nRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython) (1.26.4)\nRequirement already satisfied: mkl_fft in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (1.3.8)\nRequirement already satisfied: mkl_random in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (1.2.4)\nRequirement already satisfied: mkl_umath in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (0.1.1)\nRequirement already satisfied: mkl in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (2025.0.1)\nRequirement already satisfied: tbb4py in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (2022.0.0)\nRequirement already satisfied: mkl-service in /usr/local/lib/python3.10/dist-packages (from numpy->biopython) (2.4.1)\nRequirement already satisfied: intel-openmp>=2024 in /usr/local/lib/python3.10/dist-packages (from mkl->numpy->biopython) (2024.2.0)\nRequirement already satisfied: tbb==2022.* in /usr/local/lib/python3.10/dist-packages (from mkl->numpy->biopython) (2022.0.0)\nRequirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.10/dist-packages (from tbb==2022.*->mkl->numpy->biopython) (1.2.0)\nRequirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.10/dist-packages (from mkl_umath->numpy->biopython) (2024.2.0)\nRequirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.10/dist-packages (from intel-openmp>=2024->mkl->numpy->biopython) (2024.2.0)\nDownloading biopython-1.85-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.2/3.2 MB\u001b[0m \u001b[31m64.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m\n\u001b[?25hInstalling collected packages: biopython\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\nprotenix 0.4.6 requires biotite==1.0.1, which is not installed.\nprotenix 0.4.6 requires deepspeed>=0.15.1, which is not installed.\nprotenix 0.4.6 requires icecream, which is not installed.\nprotenix 0.4.6 requires ipdb, which is not installed.\nprotenix 0.4.6 requires ml_collections, which is not installed.\nprotenix 0.4.6 requires modelcif==0.7, which is not installed.\nprotenix 0.4.6 requires py3Dmol, which is not installed.\nprotenix 0.4.6 requires rdkit, which is not installed.\nprotenix 0.4.6 requires scikit-learn-extra, which is not installed.\nprotenix 0.4.6 requires biopython==1.83, but you have biopython 1.85 which is incompatible.\nprotenix 0.4.6 requires matplotlib==3.9.2, but you have matplotlib 3.7.5 which is incompatible.\nprotenix 0.4.6 requires numpy==1.26.3, but you have numpy 1.26.4 which is incompatible.\nprotenix 0.4.6 requires protobuf==3.20.2, but you have protobuf 3.20.3 which is incompatible.\nprotenix 0.4.6 requires torch==2.3.1, but you have torch 2.5.1+cu121 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed biopython-1.85\nCollecting ml-collections\n  Downloading ml_collections-1.1.0-py3-none-any.whl.metadata (22 kB)\nRequirement already satisfied: absl-py in /usr/local/lib/python3.10/dist-packages (from ml-collections) (1.4.0)\nRequirement already satisfied: PyYAML in /usr/local/lib/python3.10/dist-packages (from ml-collections) (6.0.2)\nDownloading ml_collections-1.1.0-py3-none-any.whl (76 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m76.7/76.7 kB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hInstalling collected packages: ml-collections\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\nprotenix 0.4.6 requires biotite==1.0.1, which is not installed.\nprotenix 0.4.6 requires deepspeed>=0.15.1, which is not installed.\nprotenix 0.4.6 requires icecream, which is not installed.\nprotenix 0.4.6 requires ipdb, which is not installed.\nprotenix 0.4.6 requires modelcif==0.7, which is not installed.\nprotenix 0.4.6 requires py3Dmol, which is not installed.\nprotenix 0.4.6 requires rdkit, which is not installed.\nprotenix 0.4.6 requires scikit-learn-extra, which is not installed.\nprotenix 0.4.6 requires biopython==1.83, but you have biopython 1.85 which is incompatible.\nprotenix 0.4.6 requires matplotlib==3.9.2, but you have matplotlib 3.7.5 which is incompatible.\nprotenix 0.4.6 requires numpy==1.26.3, but you have numpy 1.26.4 which is incompatible.\nprotenix 0.4.6 requires protobuf==3.20.2, but you have protobuf 3.20.3 which is incompatible.\nprotenix 0.4.6 requires torch==2.3.1, but you have torch 2.5.1+cu121 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed ml-collections-1.1.0\nCollecting biotite==1.0.1\n  Downloading biotite-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.2 kB)\nCollecting biotraj<2.0,>=1.0 (from biotite==1.0.1)\n  Downloading biotraj-1.2.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (32 kB)\nRequirement already satisfied: msgpack>=0.5.6 in /usr/local/lib/python3.10/dist-packages (from biotite==1.0.1) (1.1.0)\nRequirement already satisfied: networkx>=2.0 in /usr/local/lib/python3.10/dist-packages (from biotite==1.0.1) (3.4.2)\nRequirement already satisfied: numpy>=1.25 in /usr/local/lib/python3.10/dist-packages (from biotite==1.0.1) (1.26.4)\nRequirement already satisfied: requests>=2.12 in /usr/local/lib/python3.10/dist-packages (from biotite==1.0.1) (2.32.3)\nRequirement already satisfied: scipy>=1.13 in /usr/local/lib/python3.10/dist-packages (from biotraj<2.0,>=1.0->biotite==1.0.1) (1.13.1)\nRequirement already satisfied: mkl_fft in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (1.3.8)\nRequirement already satisfied: mkl_random in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (1.2.4)\nRequirement already satisfied: mkl_umath in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (0.1.1)\nRequirement already satisfied: mkl in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (2025.0.1)\nRequirement already satisfied: tbb4py in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (2022.0.0)\nRequirement already satisfied: mkl-service in /usr/local/lib/python3.10/dist-packages (from numpy>=1.25->biotite==1.0.1) (2.4.1)\nRequirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.12->biotite==1.0.1) (3.4.1)\nRequirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.12->biotite==1.0.1) (3.10)\nRequirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.12->biotite==1.0.1) (2.3.0)\nRequirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.12->biotite==1.0.1) (2025.1.31)\nRequirement already satisfied: intel-openmp>=2024 in /usr/local/lib/python3.10/dist-packages (from mkl->numpy>=1.25->biotite==1.0.1) (2024.2.0)\nRequirement already satisfied: tbb==2022.* in /usr/local/lib/python3.10/dist-packages (from mkl->numpy>=1.25->biotite==1.0.1) (2022.0.0)\nRequirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.10/dist-packages (from tbb==2022.*->mkl->numpy>=1.25->biotite==1.0.1) (1.2.0)\nRequirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.10/dist-packages (from mkl_umath->numpy>=1.25->biotite==1.0.1) (2024.2.0)\nRequirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.10/dist-packages (from intel-openmp>=2024->mkl->numpy>=1.25->biotite==1.0.1) (2024.2.0)\nDownloading biotite-1.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (33.9 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m33.9/33.9 MB\u001b[0m \u001b[31m55.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m0:01\u001b[0mm\n\u001b[?25hDownloading biotraj-1.2.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.1 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m69.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hInstalling collected packages: biotraj, biotite\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\nprotenix 0.4.6 requires deepspeed>=0.15.1, which is not installed.\nprotenix 0.4.6 requires icecream, which is not installed.\nprotenix 0.4.6 requires ipdb, which is not installed.\nprotenix 0.4.6 requires modelcif==0.7, which is not installed.\nprotenix 0.4.6 requires py3Dmol, which is not installed.\nprotenix 0.4.6 requires rdkit, which is not installed.\nprotenix 0.4.6 requires scikit-learn-extra, which is not installed.\nprotenix 0.4.6 requires biopython==1.83, but you have biopython 1.85 which is incompatible.\nprotenix 0.4.6 requires matplotlib==3.9.2, but you have matplotlib 3.7.5 which is incompatible.\nprotenix 0.4.6 requires numpy==1.26.3, but you have numpy 1.26.4 which is incompatible.\nprotenix 0.4.6 requires protobuf==3.20.2, but you have protobuf 3.20.3 which is incompatible.\nprotenix 0.4.6 requires torch==2.3.1, but you have torch 2.5.1+cu121 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed biotite-1.0.1 biotraj-1.2.2\nCollecting rdkit\n  Downloading rdkit-2025.3.2-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (4.0 kB)\nRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from rdkit) (1.26.4)\nRequirement already satisfied: Pillow in /usr/local/lib/python3.10/dist-packages (from rdkit) (11.0.0)\nRequirement already satisfied: mkl_fft in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (1.3.8)\nRequirement already satisfied: mkl_random in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (1.2.4)\nRequirement already satisfied: mkl_umath in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (0.1.1)\nRequirement already satisfied: mkl in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (2025.0.1)\nRequirement already satisfied: tbb4py in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (2022.0.0)\nRequirement already satisfied: mkl-service in /usr/local/lib/python3.10/dist-packages (from numpy->rdkit) (2.4.1)\nRequirement already satisfied: intel-openmp>=2024 in /usr/local/lib/python3.10/dist-packages (from mkl->numpy->rdkit) (2024.2.0)\nRequirement already satisfied: tbb==2022.* in /usr/local/lib/python3.10/dist-packages (from mkl->numpy->rdkit) (2022.0.0)\nRequirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.10/dist-packages (from tbb==2022.*->mkl->numpy->rdkit) (1.2.0)\nRequirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.10/dist-packages (from mkl_umath->numpy->rdkit) (2024.2.0)\nRequirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.10/dist-packages (from intel-openmp>=2024->mkl->numpy->rdkit) (2024.2.0)\nDownloading rdkit-2025.3.2-cp310-cp310-manylinux_2_28_x86_64.whl (35.2 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m35.2/35.2 MB\u001b[0m \u001b[31m50.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m00:01\u001b[0m\n\u001b[?25hInstalling collected packages: rdkit\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\nprotenix 0.4.6 requires deepspeed>=0.15.1, which is not installed.\nprotenix 0.4.6 requires icecream, which is not installed.\nprotenix 0.4.6 requires ipdb, which is not installed.\nprotenix 0.4.6 requires modelcif==0.7, which is not installed.\nprotenix 0.4.6 requires py3Dmol, which is not installed.\nprotenix 0.4.6 requires scikit-learn-extra, which is not installed.\nprotenix 0.4.6 requires biopython==1.83, but you have biopython 1.85 which is incompatible.\nprotenix 0.4.6 requires matplotlib==3.9.2, but you have matplotlib 3.7.5 which is incompatible.\nprotenix 0.4.6 requires numpy==1.26.3, but you have numpy 1.26.4 which is incompatible.\nprotenix 0.4.6 requires protobuf==3.20.2, but you have protobuf 3.20.3 which is incompatible.\nprotenix 0.4.6 requires torch==2.3.1, but you have torch 2.5.1+cu121 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed rdkit-2025.3.2\n","output_type":"stream"}],"execution_count":4},{"cell_type":"code","source":"! mkdir /af3-dev \n! ln -s /kaggle/input/protenix-checkpoints /af3-dev/release_data\n! ls /af3-dev/release_data/","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:14.080698Z","iopub.execute_input":"2025-05-14T02:15:14.081025Z","iopub.status.idle":"2025-05-14T02:15:14.436581Z","shell.execute_reply.started":"2025-05-14T02:15:14.081002Z","shell.execute_reply":"2025-05-14T02:15:14.435572Z"}},"outputs":[{"name":"stdout","text":"components.v20240608.cif\t\tmodel_v0.2.0.pt\ncomponents.v20240608.cif.rdkit_mol.pkl\n","output_type":"stream"}],"execution_count":5},{"cell_type":"markdown","source":"## Helper scripts","metadata":{}},{"cell_type":"code","source":"import Bio\n\nfrom copy import deepcopy\n\nimport pandas as pd\nfrom Bio.PDB import Atom, Model, Chain, Residue, Structure, PDBParser\nfrom Bio import SeqIO\nimport os, sys\nimport re\nimport numpy as np\nimport torch\n\nimport matplotlib\nimport matplotlib.pyplot as plt\nfrom tqdm import tqdm\nimport time\ntime0=time.time()\nfrom Bio import AlignIO\nprint('IMPORT OK !!!!')","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:17.453248Z","iopub.execute_input":"2025-05-14T02:15:17.453637Z","iopub.status.idle":"2025-05-14T02:15:19.510848Z","shell.execute_reply.started":"2025-05-14T02:15:17.453606Z","shell.execute_reply":"2025-05-14T02:15:19.510032Z"}},"outputs":[{"name":"stdout","text":"IMPORT OK !!!!\n","output_type":"stream"}],"execution_count":6},{"cell_type":"code","source":"PYTHON = sys.executable\nprint('PYTHON',PYTHON)\n\nRHONET_DIR=\\\n'/kaggle/input/data-for-demo-for-rhofold-plus-with-kaggle-msa/RhoFold-main'\n#'<your downloaded rhofold repo>/RhoFold-main'\n\nUSALIGN = \\\n'/kaggle/working//USalign'\n#'<your us align path>/USalign'\n\nos.system('cp /kaggle/input/usalign/USalign /kaggle/working/')\nos.system('sudo chmod u+x /kaggle/working//USalign')\nsys.path.append(RHONET_DIR)\n\n\nDATA_KAGGLE_DIR = '/kaggle/input/stanford-rna-3d-folding'\n\n\n# helper ----\n# 딕셔너리 <-> 키 매핑\nclass dotdict(dict):\n\t__setattr__ = dict.__setitem__\n\t__delattr__ = dict.__delitem__\n\n\tdef __getattr__(self, name):\n\t\ttry:\n\t\t\treturn self[name]\n\t\texcept KeyError:\n\t\t\traise AttributeError(name)\n\n# visualisation helper ----\ndef set_aspect_equal(ax):\n\tx_limits = ax.get_xlim()\n\ty_limits = ax.get_ylim()\n\tz_limits = ax.get_zlim()\n\n\t# Compute the mean of each axis\n\tx_middle = np.mean(x_limits)\n\ty_middle = np.mean(y_limits)\n\tz_middle = np.mean(z_limits)\n\n\t# Compute the max range across all axes\n\tmax_range = max(x_limits[1] - x_limits[0],\n\t\t\t\t\ty_limits[1] - y_limits[0],\n\t\t\t\t\tz_limits[1] - z_limits[0]) / 2.0\n\n\t# Set the new limits to ensure equal scaling\n\tax.set_xlim(x_middle - max_range, x_middle + max_range)\n\tax.set_ylim(y_middle - max_range, y_middle + max_range)\n\tax.set_zlim(z_middle - max_range, z_middle + max_range)\n\n\n\n\n# xyz df helper --------------------\ndef get_truth_df(target_id):\n    truth_df = LABEL_DF[LABEL_DF['target_id'] == target_id]\n    truth_df = truth_df.reset_index(drop=True)\n    return truth_df\n\ndef parse_output_to_df(output, seq, target_id):\n    df = []\n    chain_data = []\n    for i, res in enumerate(seq):\n        d=dict(ID = target_id,\n                    resname=res,\n                    resid=i+1)\n        for n in range(len(output)):\n            d={**d, f'x_{n+1}': round(output[n,i,0].item(),3),\n                     f'y_{n+1}': round(output[n,i,1].item(),3),\n                     f'z_{n+1}': round(output[n,i,2].item(),3)}\n        chain_data.append(d)\n\n    if len(chain_data)!=0:\n        chain_df = pd.DataFrame(chain_data)\n        df.append(chain_df)\n        ##print(chain_df)\n    return df\n\ndef parse_pdb_to_df(pdb_file, target_id):\n    parser = PDBParser()\n    structure = parser.get_structure('', pdb_file)\n\n    df = []\n    for model in structure:\n        for chain in model:\n            print(chain)\n            chain_data = []\n            for residue in chain:\n                # print(residue)\n                if residue.get_resname() in ['A', 'U', 'G', 'C']:\n                    # Check if the residue has a C1' atom\n                    if 'C1\\'' in residue:\n                        atom = residue['C1\\'']\n                        xyz = atom.get_coord()\n                        resname = residue.get_resname()\n                        resid = residue.get_id()[1]\n\n                        #todo detect discontinous: resid = prev_resid+1\n                        #ID\tresname\tresid\tx_1\ty_1\tz_1\n                        chain_data.append(dict(\n                            ID = target_id+'_'+str(resid),\n                            resname=resname,\n                            resid=resid,\n                            x_1=xyz[0],\n                            y_1=xyz[1],\n                            z_1=xyz[2],\n                        ))\n                        ##print(f\"Residue {resname} {resid}, Atom: {atom.get_name()}, xyz: {xyz}\")\n\n            if len(chain_data)!=0:\n                chain_df = pd.DataFrame(chain_data)\n                df.append(chain_df)\n                ##print(chain_df)\n    return df\n\n# usalign helper --------------------\ndef write_target_line(\n    atom_name, atom_serial, residue_name, chain_id, residue_num, x_coord, y_coord, z_coord, occupancy=1.0, b_factor=0.0, atom_type='P'\n):\n    \"\"\"\n    Writes a single line of PDB format based on provided atom information.\n\n    Args:\n        atom_name (str): Name of the atom (e.g., \"N\", \"CA\").\n        atom_serial (int): Atom serial number.\n        residue_name (str): Residue name (e.g., \"ALA\").\n        chain_id (str): Chain identifier.\n        residue_num (int): Residue number.\n        x_coord (float): X coordinate.\n        y_coord (float): Y coordinate.\n        z_coord (float): Z coordinate.\n        occupancy (float, optional): Occupancy value (default: 1.0).\n        b_factor (float, optional): B-factor value (default: 0.0).\n\n    Returns:\n        str: A single line of PDB string.\n    \"\"\"\n    return f'ATOM  {atom_serial:>5d}  {atom_name:<5s} {residue_name:<3s} {residue_num:>3d}    {x_coord:>8.3f}{y_coord:>8.3f}{z_coord:>8.3f}{occupancy:>6.2f}{b_factor:>6.2f}           {atom_type}\\n'\n\ndef write_xyz_to_pdb(df, pdb_file, xyz_id = 1):\n    resolved_cnt = 0\n    with open(pdb_file, 'w') as target_file:\n        for _, row in df.iterrows():\n            x_coord = row[f'x_{xyz_id}']\n            y_coord = row[f'y_{xyz_id}']\n            z_coord = row[f'z_{xyz_id}']\n\n            if x_coord > -1e17 and y_coord > -1e17 and z_coord > -1e17:\n                resolved_cnt += 1\n                target_line = write_target_line(\n                    atom_name=\"C1'\",\n                    atom_serial=int(row['resid']),\n                    residue_name=row['resname'],\n                    chain_id='0',\n                    residue_num=int(row['resid']),\n                    x_coord=x_coord,\n                    y_coord=y_coord,\n                    z_coord=z_coord,\n                    atom_type='C',\n                )\n                target_file.write(target_line)\n    return resolved_cnt\n\ndef parse_usalign_for_tm_score(output):\n    # Extract TM-score based on length of reference structure (second)\n    tm_score_match = re.findall(r'TM-score=\\s+([\\d.]+)', output)[1]\n    if not tm_score_match:\n        raise ValueError('No TM score found')\n    return float(tm_score_match)\n\ndef parse_usalign_for_transform(output):\n    # Locate the rotation matrix section\n    matrix_lines = []\n    found_matrix = False\n\n    for line in output.splitlines():\n        if \"The rotation matrix to rotate Structure_1 to Structure_2\" in line:\n            found_matrix = True\n        elif found_matrix and re.match(r'^\\d+\\s+[-\\d.]+\\s+[-\\d.]+\\s+[-\\d.]+\\s+[-\\d.]+$', line):\n            matrix_lines.append(line)\n        elif found_matrix and not line.strip():\n            break  # Stop parsing if an empty line is encountered after the matrix\n\n    # Parse the rotation matrix values\n    rotation_matrix = []\n    for line in matrix_lines:\n        parts = line.split()\n        row_values = list(map(float, parts[1:]))  # Skip the first column (index)\n        rotation_matrix.append(row_values)\n\n    return np.array(rotation_matrix)\n\ndef call_usalign(predict_df, truth_df, verbose=1):\n    truth_pdb = '~truth.pdb'\n    predict_pdb = '~predict.pdb'\n    write_xyz_to_pdb(predict_df, predict_pdb, xyz_id=1)\n    write_xyz_to_pdb(truth_df, truth_pdb, xyz_id=1)\n\n    command = f'{USALIGN} {predict_pdb} {truth_pdb} -atom \" C1\\'\" -m -'\n    output = os.popen(command).read()\n    if verbose==1:\n        print(output)\n    tm_score = parse_usalign_for_tm_score(output)\n    transform = parse_usalign_for_transform(output)\n    return tm_score, transform\n\nprint('HELPER OK!!!')","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:18:45.504501Z","iopub.execute_input":"2025-05-14T02:18:45.505006Z","iopub.status.idle":"2025-05-14T02:18:45.630154Z","shell.execute_reply.started":"2025-05-14T02:18:45.504964Z","shell.execute_reply":"2025-05-14T02:18:45.629389Z"}},"outputs":[{"name":"stdout","text":"PYTHON /usr/bin/python3\nHELPER OK!!!\n","output_type":"stream"}],"execution_count":17},{"cell_type":"code","source":"if MODEL_TYPE=='protenix':\n    \n    \n    from runner.batch_inference import get_default_runner\n    from runner.inference import update_inference_configs, InferenceRunner\n\n    from protenix.data.infer_data_pipeline import InferenceDataset\n\n    np.random.seed(0)\n    torch.random.manual_seed(0)\n    torch.cuda.manual_seed_all(0)\n\nclass DictDataset(InferenceDataset):\n    def __init__(\n        self,\n        seq_list: list,\n        dump_dir: str,\n        id_list: list = None,\n        msa_root_1: str = \"/kaggle/input/stanford-rna-3d-folding/MSA\",\n        msa_root_2: str = \"/kaggle/input/stanford-rna-3d-folding/MSA_v2\",\n    ) -> None:\n        self.dump_dir = dump_dir\n        self.msa_roots = [msa_root_2, msa_root_1]  # 우선순위: v2 > v1\n        self.use_msa = True  # Protenix 내부에서 참조하므로 반드시 필요\n\n        if isinstance(id_list, type(None)):\n            self.inputs = [{\"sequences\": [{\"rnaSequence\": {\"sequence\": seq, \"count\": 1}}], \"name\": \"query\"}\n                           for seq in seq_list]\n        else:\n            self.inputs = [{\"sequences\": [{\"rnaSequence\": {\"sequence\": seq, \"count\": 1}}], \"name\": i}\n                           for i, seq in zip(id_list, seq_list)]\n\n    def __getitem__(self, index):\n        data, atom_array, error_msg = super().__getitem__(index)\n        target_id = self.inputs[index][\"name\"]\n\n        msa_tensor = None\n        for root in self.msa_roots:\n            msa_path = os.path.join(root, f\"{target_id}.MSA.fasta\")\n            if os.path.exists(msa_path):\n                try:\n                    msa_tensor = self.load_msa_fasta(msa_path)\n                    print(f\"✅ Loaded MSA for {target_id} from {root} | shape={msa_tensor.shape}\")\n                    break\n                except Exception as e:\n                    print(f\"⚠️ Error loading MSA for {target_id}: {e}\")\n\n        if msa_tensor is not None and msa_tensor.shape[0] >= 10:\n            if \"input_feature_dict\" not in data:\n                data[\"input_feature_dict\"] = {}\n            data[\"input_feature_dict\"][\"msa_feat\"] = msa_tensor\n            print(f\"✅ Injected msa_feat for {target_id} | shape={msa_tensor.shape}\")\n        else:\n            print(f\"⚠️ Skipped msa_feat for {target_id} | depth insufficient or not found\")\n\n        return data, atom_array, error_msg\n\n    def load_msa_fasta(self, msa_path):\n        from Bio import AlignIO\n        alignment = AlignIO.read(msa_path, \"fasta\")\n        alignment = alignment[:64]  # optional: top 64 rows only\n        L = len(alignment[0])\n        N = len(alignment)\n\n        # RNA base vocabulary mapping\n        alphabet = {'A': 0, 'C': 1, 'G': 2, 'U': 3, '-': 4}\n        msa_tensor = torch.zeros((N, L), dtype=torch.long)\n        for i, record in enumerate(alignment):\n            msa_tensor[i] = torch.tensor([\n                alphabet.get(residue.upper(), 4)  # default to gap if unknown\n                for residue in record.seq\n            ])\n        return msa_tensor\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:46:32.380799Z","iopub.execute_input":"2025-05-14T02:46:32.381092Z","iopub.status.idle":"2025-05-14T02:46:32.392705Z","shell.execute_reply.started":"2025-05-14T02:46:32.381069Z","shell.execute_reply":"2025-05-14T02:46:32.392002Z"}},"outputs":[],"execution_count":25},{"cell_type":"code","source":"if MODEL_TYPE=='protenix':\n\n    from configs.configs_base import configs as configs_base\n    from configs.configs_data import data_configs\n    from configs.configs_inference import inference_configs\n    from protenix.config.config import parse_configs\n\n    configs_base[\"use_deepspeed_evo_attention\"] = (\n    os.environ.get(\"USE_DEEPSPEED_EVO_ATTENTION\", False) == \"true\")\n    configs_base[\"model\"][\"N_cycle\"] = 10 #10\n    configs_base[\"sample_diffusion\"][\"N_sample\"] = (1 if VALIDATION else 5)\n    configs_base[\"sample_diffusion\"][\"N_step\"] = 200\n    inference_configs['load_checkpoint_path']='/kaggle/input/protenix-checkpoints/model_v0.2.0.pt'\n    configs = {**configs_base, **{\"data\": data_configs}, **inference_configs}\n\n    configs = parse_configs(\n            configs=configs,\n            fill_required_with_null=True,\n        )\n    \n    runner=InferenceRunner(configs)\n    # print(type(runner.model))          # 클래스 이름\n    # print(runner.model)                # 전체 구조 요약","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:15:29.527953Z","iopub.execute_input":"2025-05-14T02:15:29.528521Z","iopub.status.idle":"2025-05-14T02:15:52.294169Z","shell.execute_reply.started":"2025-05-14T02:15:29.528492Z","shell.execute_reply":"2025-05-14T02:15:52.293241Z"}},"outputs":[{"name":"stdout","text":"train scheduler 16.0\ninference scheduler 16.0\nDiffusion Module has 16.0\n","output_type":"stream"},{"name":"stderr","text":"/usr/local/lib/python3.10/dist-packages/runner/inference.py:107: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.\n  checkpoint = torch.load(checkpoint_path, self.device)\n","output_type":"stream"},{"name":"stdout","text":"<class 'protenix.model.protenix.Protenix'>\nProtenix(\n  (input_embedder): InputFeatureEmbedder(\n    (atom_attention_encoder): AtomAttentionEncoder(\n      (linear_no_bias_f): Linear(in_features=389, out_features=128, bias=False)\n      (linear_no_bias_d): Linear(in_features=3, out_features=16, bias=False)\n      (linear_no_bias_invd): Linear(in_features=1, out_features=16, bias=False)\n      (linear_no_bias_v): Linear(in_features=1, out_features=16, bias=False)\n      (linear_no_bias_cl): Linear(in_features=128, out_features=16, bias=False)\n      (linear_no_bias_cm): Linear(in_features=128, out_features=16, bias=False)\n      (small_mlp): Sequential(\n        (0): ReLU()\n        (1): Linear(in_features=16, out_features=16, bias=False)\n        (2): ReLU()\n        (3): Linear(in_features=16, out_features=16, bias=False)\n        (4): ReLU()\n        (5): Linear(in_features=16, out_features=16, bias=False)\n      )\n      (atom_transformer): AtomTransformer(\n        (diffusion_transformer): DiffusionTransformer(\n          (blocks): ModuleList(\n            (0-2): 3 x DiffusionTransformerBlock(\n              (attention_pair_bias): AttentionPairBias(\n                (layernorm_a): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_a_last): BiasInitLinear(in_features=128, out_features=128, bias=True)\n                (attention): Attention(\n                  (linear_q): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_k): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_v): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_o): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_g): Linear(in_features=128, out_features=128, bias=False)\n                  (sigmoid): Sigmoid()\n                )\n                (layernorm_z): OpenFoldLayerNorm()\n                (linear_nobias_z): Linear(in_features=16, out_features=4, bias=False)\n              )\n              (conditioned_transition_block): ConditionedTransitionBlock(\n                (adaln): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_nobias_a1): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_a2): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_b): Linear(in_features=256, out_features=128, bias=False)\n                (linear_s): BiasInitLinear(in_features=128, out_features=128, bias=True)\n              )\n            )\n          )\n        )\n      )\n      (linear_no_bias_q): Linear(in_features=128, out_features=384, bias=False)\n    )\n  )\n  (relative_position_encoding): RelativePositionEncoding(\n    (linear_no_bias): Linear(in_features=139, out_features=128, bias=False)\n  )\n  (template_embedder): TemplateEmbedder(\n    (linear_no_bias_z): Linear(in_features=128, out_features=64, bias=False)\n    (layernorm_z): OpenFoldLayerNorm()\n    (linear_no_bias_a): Linear(in_features=108, out_features=64, bias=False)\n    (pairformer_stack): PairformerStack(\n      (blocks): ModuleList()\n    )\n    (layernorm_v): OpenFoldLayerNorm()\n    (linear_no_bias_u): Linear(in_features=64, out_features=128, bias=False)\n  )\n  (msa_module): MSAModule(\n    (linear_no_bias_m): Linear(in_features=34, out_features=64, bias=False)\n    (linear_no_bias_s): Linear(in_features=449, out_features=64, bias=False)\n    (blocks): ModuleList(\n      (0-2): 3 x MSABlock(\n        (outer_product_mean_msa): OuterProductMean(\n          (layer_norm): LayerNorm((64,), eps=1e-05, elementwise_affine=True)\n          (linear_1): Linear(in_features=64, out_features=32, bias=True)\n          (linear_2): Linear(in_features=64, out_features=32, bias=True)\n          (linear_out): Linear(in_features=1024, out_features=128, bias=True)\n        )\n        (msa_stack): MSAStack(\n          (msa_pair_weighted_averaging): MSAPairWeightedAveraging(\n            (layernorm_m): OpenFoldLayerNorm()\n            (linear_no_bias_mv): Linear(in_features=64, out_features=64, bias=False)\n            (layernorm_z): OpenFoldLayerNorm()\n            (linear_no_bias_z): Linear(in_features=128, out_features=8, bias=False)\n            (linear_no_bias_mg): Linear(in_features=64, out_features=64, bias=False)\n            (softmax_w): Softmax(dim=-2)\n            (linear_no_bias_out): Linear(in_features=64, out_features=64, bias=False)\n          )\n          (dropout_row): DropoutRowwise(\n            (dropout): Dropout(p=0.15, inplace=False)\n          )\n          (transition_m): Transition(\n            (layernorm1): OpenFoldLayerNorm()\n            (linear_no_bias_a): Linear(in_features=64, out_features=256, bias=False)\n            (linear_no_bias_b): Linear(in_features=64, out_features=256, bias=False)\n            (linear_no_bias): Linear(in_features=256, out_features=64, bias=False)\n          )\n        )\n        (pair_stack): PairformerBlock(\n          (tri_mul_out): TriangleMultiplicationOutgoing(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_mul_in): TriangleMultiplicationIncoming(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_att_start): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (tri_att_end): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (dropout_row): DropoutRowwise(\n            (dropout): Dropout(p=0.25, inplace=False)\n          )\n          (pair_transition): Transition(\n            (layernorm1): OpenFoldLayerNorm()\n            (linear_no_bias_a): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias_b): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias): Linear(in_features=512, out_features=128, bias=False)\n          )\n        )\n      )\n      (3): MSABlock(\n        (outer_product_mean_msa): OuterProductMean(\n          (layer_norm): LayerNorm((64,), eps=1e-05, elementwise_affine=True)\n          (linear_1): Linear(in_features=64, out_features=32, bias=True)\n          (linear_2): Linear(in_features=64, out_features=32, bias=True)\n          (linear_out): Linear(in_features=1024, out_features=128, bias=True)\n        )\n        (pair_stack): PairformerBlock(\n          (tri_mul_out): TriangleMultiplicationOutgoing(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_mul_in): TriangleMultiplicationIncoming(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_att_start): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (tri_att_end): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (dropout_row): DropoutRowwise(\n            (dropout): Dropout(p=0.25, inplace=False)\n          )\n          (pair_transition): Transition(\n            (layernorm1): OpenFoldLayerNorm()\n            (linear_no_bias_a): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias_b): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias): Linear(in_features=512, out_features=128, bias=False)\n          )\n        )\n      )\n    )\n  )\n  (pairformer_stack): PairformerStack(\n    (blocks): ModuleList(\n      (0-47): 48 x PairformerBlock(\n        (tri_mul_out): TriangleMultiplicationOutgoing(\n          (linear_g): Linear(in_features=128, out_features=128, bias=True)\n          (linear_z): Linear(in_features=128, out_features=128, bias=True)\n          (layer_norm_in): OpenFoldLayerNorm()\n          (layer_norm_out): OpenFoldLayerNorm()\n          (sigmoid): Sigmoid()\n          (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n          (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n          (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n          (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n        )\n        (tri_mul_in): TriangleMultiplicationIncoming(\n          (linear_g): Linear(in_features=128, out_features=128, bias=True)\n          (linear_z): Linear(in_features=128, out_features=128, bias=True)\n          (layer_norm_in): OpenFoldLayerNorm()\n          (layer_norm_out): OpenFoldLayerNorm()\n          (sigmoid): Sigmoid()\n          (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n          (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n          (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n          (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n        )\n        (tri_att_start): TriangleAttention(\n          (layer_norm): OpenFoldLayerNorm()\n          (linear): Linear(in_features=128, out_features=4, bias=False)\n          (mha): Attention(\n            (linear_q): Linear(in_features=128, out_features=128, bias=False)\n            (linear_k): Linear(in_features=128, out_features=128, bias=False)\n            (linear_v): Linear(in_features=128, out_features=128, bias=False)\n            (linear_o): Linear(in_features=128, out_features=128, bias=True)\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (sigmoid): Sigmoid()\n          )\n        )\n        (tri_att_end): TriangleAttention(\n          (layer_norm): OpenFoldLayerNorm()\n          (linear): Linear(in_features=128, out_features=4, bias=False)\n          (mha): Attention(\n            (linear_q): Linear(in_features=128, out_features=128, bias=False)\n            (linear_k): Linear(in_features=128, out_features=128, bias=False)\n            (linear_v): Linear(in_features=128, out_features=128, bias=False)\n            (linear_o): Linear(in_features=128, out_features=128, bias=True)\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (sigmoid): Sigmoid()\n          )\n        )\n        (dropout_row): DropoutRowwise(\n          (dropout): Dropout(p=0.25, inplace=False)\n        )\n        (pair_transition): Transition(\n          (layernorm1): OpenFoldLayerNorm()\n          (linear_no_bias_a): Linear(in_features=128, out_features=512, bias=False)\n          (linear_no_bias_b): Linear(in_features=128, out_features=512, bias=False)\n          (linear_no_bias): Linear(in_features=512, out_features=128, bias=False)\n        )\n        (attention_pair_bias): AttentionPairBias(\n          (layernorm_a): OpenFoldLayerNorm()\n          (attention): Attention(\n            (linear_q): Linear(in_features=384, out_features=384, bias=True)\n            (linear_k): Linear(in_features=384, out_features=384, bias=False)\n            (linear_v): Linear(in_features=384, out_features=384, bias=False)\n            (linear_o): Linear(in_features=384, out_features=384, bias=False)\n            (linear_g): Linear(in_features=384, out_features=384, bias=False)\n            (sigmoid): Sigmoid()\n          )\n          (layernorm_z): OpenFoldLayerNorm()\n          (linear_nobias_z): Linear(in_features=128, out_features=16, bias=False)\n        )\n        (single_transition): Transition(\n          (layernorm1): OpenFoldLayerNorm()\n          (linear_no_bias_a): Linear(in_features=384, out_features=1536, bias=False)\n          (linear_no_bias_b): Linear(in_features=384, out_features=1536, bias=False)\n          (linear_no_bias): Linear(in_features=1536, out_features=384, bias=False)\n        )\n      )\n    )\n  )\n  (diffusion_module): DiffusionModule(\n    (diffusion_conditioning): DiffusionConditioning(\n      (relpe): RelativePositionEncoding(\n        (linear_no_bias): Linear(in_features=139, out_features=128, bias=False)\n      )\n      (layernorm_z): OpenFoldLayerNorm()\n      (linear_no_bias_z): Linear(in_features=256, out_features=128, bias=False)\n      (transition_z1): Transition(\n        (layernorm1): OpenFoldLayerNorm()\n        (linear_no_bias_a): Linear(in_features=128, out_features=256, bias=False)\n        (linear_no_bias_b): Linear(in_features=128, out_features=256, bias=False)\n        (linear_no_bias): Linear(in_features=256, out_features=128, bias=False)\n      )\n      (transition_z2): Transition(\n        (layernorm1): OpenFoldLayerNorm()\n        (linear_no_bias_a): Linear(in_features=128, out_features=256, bias=False)\n        (linear_no_bias_b): Linear(in_features=128, out_features=256, bias=False)\n        (linear_no_bias): Linear(in_features=256, out_features=128, bias=False)\n      )\n      (layernorm_s): OpenFoldLayerNorm()\n      (linear_no_bias_s): Linear(in_features=833, out_features=384, bias=False)\n      (fourier_embedding): FourierEmbedding()\n      (layernorm_n): OpenFoldLayerNorm()\n      (linear_no_bias_n): Linear(in_features=256, out_features=384, bias=False)\n      (transition_s1): Transition(\n        (layernorm1): OpenFoldLayerNorm()\n        (linear_no_bias_a): Linear(in_features=384, out_features=768, bias=False)\n        (linear_no_bias_b): Linear(in_features=384, out_features=768, bias=False)\n        (linear_no_bias): Linear(in_features=768, out_features=384, bias=False)\n      )\n      (transition_s2): Transition(\n        (layernorm1): OpenFoldLayerNorm()\n        (linear_no_bias_a): Linear(in_features=384, out_features=768, bias=False)\n        (linear_no_bias_b): Linear(in_features=384, out_features=768, bias=False)\n        (linear_no_bias): Linear(in_features=768, out_features=384, bias=False)\n      )\n    )\n    (atom_attention_encoder): AtomAttentionEncoder(\n      (linear_no_bias_f): Linear(in_features=389, out_features=128, bias=False)\n      (linear_no_bias_d): Linear(in_features=3, out_features=16, bias=False)\n      (linear_no_bias_invd): Linear(in_features=1, out_features=16, bias=False)\n      (linear_no_bias_v): Linear(in_features=1, out_features=16, bias=False)\n      (layernorm_s): OpenFoldLayerNorm()\n      (linear_no_bias_s): Linear(in_features=384, out_features=128, bias=False)\n      (layernorm_z): OpenFoldLayerNorm()\n      (linear_no_bias_z): Linear(in_features=128, out_features=16, bias=False)\n      (linear_no_bias_r): Linear(in_features=3, out_features=128, bias=False)\n      (linear_no_bias_cl): Linear(in_features=128, out_features=16, bias=False)\n      (linear_no_bias_cm): Linear(in_features=128, out_features=16, bias=False)\n      (small_mlp): Sequential(\n        (0): ReLU()\n        (1): Linear(in_features=16, out_features=16, bias=False)\n        (2): ReLU()\n        (3): Linear(in_features=16, out_features=16, bias=False)\n        (4): ReLU()\n        (5): Linear(in_features=16, out_features=16, bias=False)\n      )\n      (atom_transformer): AtomTransformer(\n        (diffusion_transformer): DiffusionTransformer(\n          (blocks): ModuleList(\n            (0-2): 3 x DiffusionTransformerBlock(\n              (attention_pair_bias): AttentionPairBias(\n                (layernorm_a): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_a_last): BiasInitLinear(in_features=128, out_features=128, bias=True)\n                (attention): Attention(\n                  (linear_q): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_k): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_v): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_o): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_g): Linear(in_features=128, out_features=128, bias=False)\n                  (sigmoid): Sigmoid()\n                )\n                (layernorm_z): OpenFoldLayerNorm()\n                (linear_nobias_z): Linear(in_features=16, out_features=4, bias=False)\n              )\n              (conditioned_transition_block): ConditionedTransitionBlock(\n                (adaln): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_nobias_a1): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_a2): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_b): Linear(in_features=256, out_features=128, bias=False)\n                (linear_s): BiasInitLinear(in_features=128, out_features=128, bias=True)\n              )\n            )\n          )\n        )\n      )\n      (linear_no_bias_q): Linear(in_features=128, out_features=768, bias=False)\n    )\n    (layernorm_s): OpenFoldLayerNorm()\n    (linear_no_bias_s): Linear(in_features=384, out_features=768, bias=False)\n    (diffusion_transformer): DiffusionTransformer(\n      (blocks): ModuleList(\n        (0-23): 24 x DiffusionTransformerBlock(\n          (attention_pair_bias): AttentionPairBias(\n            (layernorm_a): AdaptiveLayerNorm(\n              (layernorm_a): LayerNorm((768,), eps=1e-05, elementwise_affine=False)\n              (layernorm_s): LayerNorm((384,), eps=1e-05, elementwise_affine=True)\n              (linear_s): Linear(in_features=384, out_features=768, bias=True)\n              (linear_nobias_s): Linear(in_features=384, out_features=768, bias=False)\n            )\n            (linear_a_last): BiasInitLinear(in_features=384, out_features=768, bias=True)\n            (attention): Attention(\n              (linear_q): Linear(in_features=768, out_features=768, bias=True)\n              (linear_k): Linear(in_features=768, out_features=768, bias=False)\n              (linear_v): Linear(in_features=768, out_features=768, bias=False)\n              (linear_o): Linear(in_features=768, out_features=768, bias=False)\n              (linear_g): Linear(in_features=768, out_features=768, bias=False)\n              (sigmoid): Sigmoid()\n            )\n            (layernorm_z): OpenFoldLayerNorm()\n            (linear_nobias_z): Linear(in_features=128, out_features=16, bias=False)\n          )\n          (conditioned_transition_block): ConditionedTransitionBlock(\n            (adaln): AdaptiveLayerNorm(\n              (layernorm_a): LayerNorm((768,), eps=1e-05, elementwise_affine=False)\n              (layernorm_s): LayerNorm((384,), eps=1e-05, elementwise_affine=True)\n              (linear_s): Linear(in_features=384, out_features=768, bias=True)\n              (linear_nobias_s): Linear(in_features=384, out_features=768, bias=False)\n            )\n            (linear_nobias_a1): Linear(in_features=768, out_features=1536, bias=False)\n            (linear_nobias_a2): Linear(in_features=768, out_features=1536, bias=False)\n            (linear_nobias_b): Linear(in_features=1536, out_features=768, bias=False)\n            (linear_s): BiasInitLinear(in_features=384, out_features=768, bias=True)\n          )\n        )\n      )\n    )\n    (layernorm_a): OpenFoldLayerNorm()\n    (atom_attention_decoder): AtomAttentionDecoder(\n      (linear_no_bias_a): Linear(in_features=768, out_features=128, bias=False)\n      (layernorm_q): OpenFoldLayerNorm()\n      (linear_no_bias_out): Linear(in_features=128, out_features=3, bias=False)\n      (atom_transformer): AtomTransformer(\n        (diffusion_transformer): DiffusionTransformer(\n          (blocks): ModuleList(\n            (0-2): 3 x DiffusionTransformerBlock(\n              (attention_pair_bias): AttentionPairBias(\n                (layernorm_a): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_a_last): BiasInitLinear(in_features=128, out_features=128, bias=True)\n                (attention): Attention(\n                  (linear_q): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_k): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_v): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_o): Linear(in_features=128, out_features=128, bias=False)\n                  (linear_g): Linear(in_features=128, out_features=128, bias=False)\n                  (sigmoid): Sigmoid()\n                )\n                (layernorm_z): OpenFoldLayerNorm()\n                (linear_nobias_z): Linear(in_features=16, out_features=4, bias=False)\n              )\n              (conditioned_transition_block): ConditionedTransitionBlock(\n                (adaln): AdaptiveLayerNorm(\n                  (layernorm_a): LayerNorm((128,), eps=1e-05, elementwise_affine=False)\n                  (layernorm_s): LayerNorm((128,), eps=1e-05, elementwise_affine=True)\n                  (linear_s): Linear(in_features=128, out_features=128, bias=True)\n                  (linear_nobias_s): Linear(in_features=128, out_features=128, bias=False)\n                )\n                (linear_nobias_a1): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_a2): Linear(in_features=128, out_features=256, bias=False)\n                (linear_nobias_b): Linear(in_features=256, out_features=128, bias=False)\n                (linear_s): BiasInitLinear(in_features=128, out_features=128, bias=True)\n              )\n            )\n          )\n        )\n      )\n    )\n  )\n  (distogram_head): DistogramHead(\n    (linear): Linear(in_features=128, out_features=64, bias=True)\n  )\n  (confidence_head): ConfidenceHead(\n    (linear_no_bias_s1): Linear(in_features=449, out_features=128, bias=False)\n    (linear_no_bias_s2): Linear(in_features=449, out_features=128, bias=False)\n    (linear_no_bias_d): Linear(in_features=39, out_features=128, bias=False)\n    (pairformer_stack): PairformerStack(\n      (blocks): ModuleList(\n        (0-3): 4 x PairformerBlock(\n          (tri_mul_out): TriangleMultiplicationOutgoing(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_mul_in): TriangleMultiplicationIncoming(\n            (linear_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_z): Linear(in_features=128, out_features=128, bias=True)\n            (layer_norm_in): OpenFoldLayerNorm()\n            (layer_norm_out): OpenFoldLayerNorm()\n            (sigmoid): Sigmoid()\n            (linear_a_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_a_g): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_p): Linear(in_features=128, out_features=128, bias=True)\n            (linear_b_g): Linear(in_features=128, out_features=128, bias=True)\n          )\n          (tri_att_start): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (tri_att_end): TriangleAttention(\n            (layer_norm): OpenFoldLayerNorm()\n            (linear): Linear(in_features=128, out_features=4, bias=False)\n            (mha): Attention(\n              (linear_q): Linear(in_features=128, out_features=128, bias=False)\n              (linear_k): Linear(in_features=128, out_features=128, bias=False)\n              (linear_v): Linear(in_features=128, out_features=128, bias=False)\n              (linear_o): Linear(in_features=128, out_features=128, bias=True)\n              (linear_g): Linear(in_features=128, out_features=128, bias=True)\n              (sigmoid): Sigmoid()\n            )\n          )\n          (dropout_row): DropoutRowwise(\n            (dropout): Dropout(p=0.0, inplace=False)\n          )\n          (pair_transition): Transition(\n            (layernorm1): OpenFoldLayerNorm()\n            (linear_no_bias_a): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias_b): Linear(in_features=128, out_features=512, bias=False)\n            (linear_no_bias): Linear(in_features=512, out_features=128, bias=False)\n          )\n          (attention_pair_bias): AttentionPairBias(\n            (layernorm_a): OpenFoldLayerNorm()\n            (attention): Attention(\n              (linear_q): Linear(in_features=384, out_features=384, bias=True)\n              (linear_k): Linear(in_features=384, out_features=384, bias=False)\n              (linear_v): Linear(in_features=384, out_features=384, bias=False)\n              (linear_o): Linear(in_features=384, out_features=384, bias=False)\n              (linear_g): Linear(in_features=384, out_features=384, bias=False)\n              (sigmoid): Sigmoid()\n            )\n            (layernorm_z): OpenFoldLayerNorm()\n            (linear_nobias_z): Linear(in_features=128, out_features=16, bias=False)\n          )\n          (single_transition): Transition(\n            (layernorm1): OpenFoldLayerNorm()\n            (linear_no_bias_a): Linear(in_features=384, out_features=1536, bias=False)\n            (linear_no_bias_b): Linear(in_features=384, out_features=1536, bias=False)\n            (linear_no_bias): Linear(in_features=1536, out_features=384, bias=False)\n          )\n        )\n      )\n    )\n    (linear_no_bias_pae): Linear(in_features=128, out_features=64, bias=False)\n    (linear_no_bias_pde): Linear(in_features=128, out_features=64, bias=False)\n    (linear_no_bias_s_inputs): Linear(in_features=449, out_features=384, bias=False)\n    (linear_no_bias_s_trunk): Linear(in_features=384, out_features=384, bias=False)\n    (layernorm_s_trunk): OpenFoldLayerNorm()\n    (linear_no_bias_z_trunk): Linear(in_features=128, out_features=128, bias=False)\n    (layernorm_z_trunk): OpenFoldLayerNorm()\n    (layernorm_no_bias_z_cat): LayerNorm((256,), eps=1e-05, elementwise_affine=True)\n    (layernorm_no_bias_s_cat): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n    (linear_no_bias_z_cat): Linear(in_features=256, out_features=128, bias=False)\n    (linear_no_bias_s_cat): Linear(in_features=768, out_features=384, bias=False)\n    (pae_ln): OpenFoldLayerNorm()\n    (pde_ln): OpenFoldLayerNorm()\n    (plddt_ln): OpenFoldLayerNorm()\n    (resolved_ln): OpenFoldLayerNorm()\n  )\n  (linear_no_bias_sinit): Linear(in_features=449, out_features=384, bias=False)\n  (linear_no_bias_zinit1): Linear(in_features=384, out_features=128, bias=False)\n  (linear_no_bias_zinit2): Linear(in_features=384, out_features=128, bias=False)\n  (linear_no_bias_token_bond): Linear(in_features=1, out_features=128, bias=False)\n  (linear_no_bias_z_cycle): Linear(in_features=128, out_features=128, bias=False)\n  (linear_no_bias_s): Linear(in_features=384, out_features=384, bias=False)\n  (layernorm_z_cycle): OpenFoldLayerNorm()\n  (layernorm_s): OpenFoldLayerNorm()\n)\n","output_type":"stream"}],"execution_count":9},{"cell_type":"code","source":"if VALIDATION:\n    LABEL_DF = pd.read_csv('/kaggle/input/stanford-rna-3d-folding/train_labels.csv')\n    LABEL_DF['target_id'] = LABEL_DF['ID'].apply(lambda x: '_'.join(x.split('_')[:-1]))\n    train_df=pd.read_csv('/kaggle/input/stanford-rna-3d-folding/train_sequences.csv')\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:18:10.859035Z","iopub.execute_input":"2025-05-14T02:18:10.859347Z","iopub.status.idle":"2025-05-14T02:18:11.085375Z","shell.execute_reply.started":"2025-05-14T02:18:10.859321Z","shell.execute_reply":"2025-05-14T02:18:11.084655Z"}},"outputs":[],"execution_count":16},{"cell_type":"code","source":"if MODEL_TYPE == 'protenix' and VALIDATION:\n    import warnings\n    warnings.filterwarnings(\"ignore\")  \n\n    train_df['protenix_tm_score'] = None\n    dataset = DictDataset(\n        seq_list=train_df.sequence,\n        dump_dir='output',\n        id_list=train_df.target_id,\n        msa_root_1=\"/kaggle/input/stanford-rna-3d-folding/MSA\",\n        msa_root_2=\"/kaggle/input/stanford-rna-3d-folding/MSA_v2\",\n    )\n    num_data = len(dataset)\n\n    for i, seq in tqdm(enumerate(train_df.sequence), total=num_data):\n        target_id = train_df.loc[i, 'target_id']\n        print(f\"\\n▶ {i}: {target_id} | len={len(seq)} | TM={train_df.loc[i, 'protenix_tm_score']}\")\n\n        if not pd.isna(train_df.loc[i, 'protenix_tm_score']):\n            print(\"⏭️ TM-score already exists → skip\")\n            continue\n        if len(seq) > 300:\n            print(\"⏭️ Sequence too long → skip\")\n            continue\n\n        truth_df = get_truth_df(target_id)\n        num_valid_coords = sum(~np.isnan(truth_df.x_1))\n        if num_valid_coords < 3:\n            print(\"⏭️ Not enough ground-truth coordinates → skip\")\n            continue\n\n        data, atom_array, data_error_message = dataset[i]\n        if data_error_message != '':\n            print(f\"⏭️ Error loading data: {data_error_message}\")\n            continue\n\n        print(\"📦 input keys:\", data['input_feature_dict'].keys())\n        if \"msa_feat\" in data[\"input_feature_dict\"]:\n            msa_feat = data[\"input_feature_dict\"][\"msa_feat\"]\n            print(\"✅ msa_feat shape:\", msa_feat.shape)\n        else:\n            print(\"❌ msa_feat not present\")\n\n        new_configs = update_inference_configs(configs, data[\"N_token\"].item())\n        runner.update_model_configs(new_configs)\n\n        prediction = runner.predict(data)\n        prediction = prediction['coordinate'][:, data['input_feature_dict']['atom_to_tokatom_idx'] == 12]\n\n        print(f\"🧪 Prediction shape: {prediction.shape}, contains NaN: {torch.isnan(prediction).any().item()}\")\n\n        result = parse_output_to_df(prediction[:1], seq, target_id)[0]\n\n        try:\n            tm_score, transform = call_usalign(result, truth_df, verbose=0)\n            train_df.loc[i, 'protenix_tm_score'] = tm_score\n            print(f\"✅ TM-score: {tm_score:.4f}\")\n        except Exception as e:\n            print(f\"❌ TM-score 계산 실패 (target_id={target_id}): {e}\")\n\n        if (time.time() - time0) > (12 * 3600 - 360):\n            print(\"⏹️ 시간 제한 도달 → 종료\")\n            break\n\n    train_df.to_csv('tm_scores.csv', index=False)\n    print(\"📊 평균 TM-score:\", train_df.protenix_tm_score.mean())\n    display(train_df.protenix_tm_score.hist())\n","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2025-05-14T02:46:39.039405Z","iopub.execute_input":"2025-05-14T02:46:39.039718Z"}},"outputs":[{"name":"stderr","text":"  0%|          | 0/844 [00:00<?, ?it/s]","output_type":"stream"},{"name":"stdout","text":"\n▶ 0: 1SCL_A | len=29 | TM=None\n✅ Loaded MSA for 1SCL_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 1SCL_A | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":"  0%|          | 1/844 [00:13<3:15:35, 13.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.2296\n\n▶ 1: 1RNK_A | len=34 | TM=None\n✅ Loaded MSA for 1RNK_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([10, 34])\n✅ Injected msa_feat for 1RNK_A | shape=torch.Size([10, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 34])\n","output_type":"stream"},{"name":"stderr","text":"  0%|          | 2/844 [00:27<3:16:04, 13.97s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.1923\n\n▶ 2: 1RHT_A | len=24 | TM=None\n✅ Loaded MSA for 1RHT_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([12, 24])\n✅ Injected msa_feat for 1RHT_A | shape=torch.Size([12, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([12, 24])\n","output_type":"stream"},{"name":"stderr","text":"  0%|          | 3/844 [00:41<3:15:48, 13.97s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.1576\n\n▶ 3: 1HLX_A | len=20 | TM=None\n✅ Loaded MSA for 1HLX_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([11, 20])\n✅ Injected msa_feat for 1HLX_A | shape=torch.Size([11, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 20])\n","output_type":"stream"},{"name":"stderr","text":"  0%|          | 4/844 [00:55<3:14:26, 13.89s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.2685\n\n▶ 4: 1HMH_E | len=34 | TM=None\n✅ Loaded MSA for 1HMH_E from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1HMH_E | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 5/844 [01:09<3:14:39, 13.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.3225\n\n▶ 5: 1RNG_A | len=12 | TM=None\n✅ Loaded MSA for 1RNG_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 12])\n⚠️ Skipped msa_feat for 1RNG_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 6/844 [01:23<3:12:05, 13.75s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.1951\n\n▶ 6: 1MME_D | len=25 | TM=None\n✅ Loaded MSA for 1MME_D from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 25])\n⚠️ Skipped msa_feat for 1MME_D | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 7/844 [01:36<3:12:38, 13.81s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 25, 3]), contains NaN: False\n✅ TM-score: 0.3339\n\n▶ 7: 1KAJ_A | len=32 | TM=None\n✅ Loaded MSA for 1KAJ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 32])\n✅ Injected msa_feat for 1KAJ_A | shape=torch.Size([64, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 32])\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 8/844 [01:50<3:11:45, 13.76s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.1496\n\n▶ 8: 1SLO_A | len=19 | TM=None\n✅ Loaded MSA for 1SLO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([26, 19])\n✅ Injected msa_feat for 1SLO_A | shape=torch.Size([26, 19])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([26, 19])\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 9/844 [02:04<3:11:17, 13.75s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.2074\n\n▶ 9: 1BIV_A | len=28 | TM=None\n✅ Loaded MSA for 1BIV_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([7, 28])\n⚠️ Skipped msa_feat for 1BIV_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  1%|          | 10/844 [02:18<3:12:19, 13.84s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 28, 3]), contains NaN: False\n✅ TM-score: 0.2686\n\n▶ 10: 1ANR_A | len=29 | TM=None\n✅ Loaded MSA for 1ANR_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 1ANR_A | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":"  1%|▏         | 11/844 [02:32<3:11:57, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.1805\n\n▶ 11: 1ZIG_A | len=12 | TM=None\n✅ Loaded MSA for 1ZIG_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 12])\n⚠️ Skipped msa_feat for 1ZIG_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  1%|▏         | 12/844 [02:45<3:11:10, 13.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.2529\n\n▶ 12: 1ZIH_A | len=12 | TM=None\n✅ Loaded MSA for 1ZIH_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 12])\n⚠️ Skipped msa_feat for 1ZIH_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 13/844 [02:59<3:10:13, 13.73s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.2987\n\n▶ 13: 1ETF_A | len=34 | TM=None\n✅ Loaded MSA for 1ETF_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([33, 34])\n✅ Injected msa_feat for 1ETF_A | shape=torch.Size([33, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([33, 34])\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 14/844 [03:13<3:11:44, 13.86s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2397\n\n▶ 14: 1ZIF_A | len=12 | TM=None\n✅ Loaded MSA for 1ZIF_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 12])\n⚠️ Skipped msa_feat for 1ZIF_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 15/844 [03:27<3:10:04, 13.76s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.2583\n\n▶ 15: 1KPD_A | len=32 | TM=None\n✅ Loaded MSA for 1KPD_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([15, 32])\n✅ Injected msa_feat for 1KPD_A | shape=torch.Size([15, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([15, 32])\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 16/844 [03:41<3:10:34, 13.81s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.2236\n\n▶ 16: 1IKD_A | len=22 | TM=None\n✅ Loaded MSA for 1IKD_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([28, 22])\n✅ Injected msa_feat for 1IKD_A | shape=torch.Size([28, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([28, 22])\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 17/844 [03:55<3:10:46, 13.84s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.1802\n\n▶ 17: 1ZDI_S | len=19 | TM=None\n✅ Loaded MSA for 1ZDI_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 19])\n⚠️ Skipped msa_feat for 1ZDI_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 18/844 [04:08<3:10:40, 13.85s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.2794\n\n▶ 18: 1AFX_A | len=12 | TM=None\n✅ Loaded MSA for 1AFX_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 12])\n⚠️ Skipped msa_feat for 1AFX_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 19/844 [04:22<3:09:30, 13.78s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.1971\n\n▶ 19: 1EBQ_A | len=29 | TM=None\n✅ Loaded MSA for 1EBQ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([38, 29])\n✅ Injected msa_feat for 1EBQ_A | shape=torch.Size([38, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([38, 29])\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 20/844 [04:36<3:09:55, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.1686\n\n▶ 20: 1EBR_A | len=30 | TM=None\n✅ Loaded MSA for 1EBR_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 30])\n✅ Injected msa_feat for 1EBR_A | shape=torch.Size([64, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 30])\n","output_type":"stream"},{"name":"stderr","text":"  2%|▏         | 21/844 [04:50<3:10:40, 13.90s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.1507\n\n▶ 21: 1ULL_A | len=35 | TM=None\n✅ Loaded MSA for 1ULL_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([8, 35])\n⚠️ Skipped msa_feat for 1ULL_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 22/844 [05:04<3:10:47, 13.93s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 35, 3]), contains NaN: False\n✅ TM-score: 0.3274\n\n▶ 22: 1KIS_B | len=16 | TM=None\n✅ Loaded MSA for 1KIS_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 16])\n⚠️ Skipped msa_feat for 1KIS_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 23/844 [05:18<3:09:51, 13.88s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 16, 3]), contains NaN: False\n✅ TM-score: 0.2380\n\n▶ 23: 1KIS_A | len=16 | TM=None\n✅ Loaded MSA for 1KIS_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 16])\n⚠️ Skipped msa_feat for 1KIS_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 24/844 [05:31<3:08:12, 13.77s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 16, 3]), contains NaN: False\n✅ TM-score: 0.2268\n\n▶ 24: 1ATO_A | len=19 | TM=None\n✅ Loaded MSA for 1ATO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([55, 19])\n✅ Injected msa_feat for 1ATO_A | shape=torch.Size([55, 19])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([55, 19])\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 25/844 [05:45<3:08:14, 13.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3002\n\n▶ 25: 1TLR_A | len=23 | TM=None\n✅ Loaded MSA for 1TLR_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([10, 23])\n✅ Injected msa_feat for 1TLR_A | shape=torch.Size([10, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 23])\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 26/844 [05:59<3:08:43, 13.84s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.1377\n\n▶ 26: 1VOP_A | len=13 | TM=None\n✅ Loaded MSA for 1VOP_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1VOP_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 27/844 [06:13<3:07:20, 13.76s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2056\n\n▶ 27: 1AQO_A | len=29 | TM=None\n✅ Loaded MSA for 1AQO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([57, 29])\n✅ Injected msa_feat for 1AQO_A | shape=torch.Size([57, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([57, 29])\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 28/844 [06:27<3:08:26, 13.86s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.2201\n\n▶ 28: 1ATV_A | len=17 | TM=None\n✅ Loaded MSA for 1ATV_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([26, 17])\n✅ Injected msa_feat for 1ATV_A | shape=torch.Size([26, 17])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([26, 17])\n","output_type":"stream"},{"name":"stderr","text":"  3%|▎         | 29/844 [06:40<3:07:40, 13.82s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.2563\n\n▶ 29: 1ATW_A | len=15 | TM=None\n✅ Loaded MSA for 1ATW_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 15])\n⚠️ Skipped msa_feat for 1ATW_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  4%|▎         | 30/844 [06:54<3:07:20, 13.81s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2984\n\n▶ 30: 1UUU_A | len=19 | TM=None\n✅ Loaded MSA for 1UUU_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 19])\n⚠️ Skipped msa_feat for 1UUU_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  4%|▎         | 31/844 [07:08<3:07:49, 13.86s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3099\n\n▶ 31: 1AUD_B | len=30 | TM=None\n✅ Loaded MSA for 1AUD_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([11, 30])\n✅ Injected msa_feat for 1AUD_B | shape=torch.Size([11, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 30])\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 32/844 [07:22<3:08:21, 13.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.1338\n\n▶ 32: 2U2A_A | len=20 | TM=None\n✅ Loaded MSA for 2U2A_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([49, 20])\n✅ Injected msa_feat for 2U2A_A | shape=torch.Size([49, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([49, 20])\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 33/844 [07:36<3:08:05, 13.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.2231\n\n▶ 33: 1A4T_A | len=15 | TM=None\n✅ Loaded MSA for 1A4T_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 15])\n⚠️ Skipped msa_feat for 1A4T_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 34/844 [07:50<3:06:45, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2999\n\n▶ 34: 1A60_A | len=44 | TM=None\n✅ Loaded MSA for 1A60_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 44])\n✅ Injected msa_feat for 1A60_A | shape=torch.Size([64, 44])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 44])\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 35/844 [08:04<3:08:17, 13.96s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 44, 3]), contains NaN: False\n✅ TM-score: 0.2167\n\n▶ 35: 1A51_A | len=41 | TM=None\n✅ Loaded MSA for 1A51_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 41])\n✅ Injected msa_feat for 1A51_A | shape=torch.Size([64, 41])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 41])\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 36/844 [08:18<3:08:20, 13.99s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 41, 3]), contains NaN: False\n✅ TM-score: 0.3229\n\n▶ 36: 2A9L_A | len=38 | TM=None\n✅ Loaded MSA for 2A9L_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([18, 38])\n✅ Injected msa_feat for 2A9L_A | shape=torch.Size([18, 38])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([18, 38])\n","output_type":"stream"},{"name":"stderr","text":"  4%|▍         | 37/844 [08:32<3:08:07, 13.99s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 38, 3]), contains NaN: False\n✅ TM-score: 0.1978\n\n▶ 37: 1A1T_B | len=20 | TM=None\n✅ Loaded MSA for 1A1T_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 20])\n⚠️ Skipped msa_feat for 1A1T_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▍         | 38/844 [08:46<3:07:21, 13.95s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.1923\n\n▶ 38: 1A9N_Q | len=24 | TM=None\n✅ Loaded MSA for 1A9N_Q from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 1A9N_Q | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":"  5%|▍         | 39/844 [09:00<3:06:40, 13.91s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.4551\n\n▶ 39: 3PHP_A | len=23 | TM=None\n✅ Loaded MSA for 3PHP_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 23])\n✅ Injected msa_feat for 3PHP_A | shape=torch.Size([64, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 23])\n","output_type":"stream"},{"name":"stderr","text":"  5%|▍         | 40/844 [09:14<3:05:57, 13.88s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.1550\n\n▶ 40: 2TPK_A | len=36 | TM=None\n✅ Loaded MSA for 2TPK_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([13, 36])\n✅ Injected msa_feat for 2TPK_A | shape=torch.Size([13, 36])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([13, 36])\n","output_type":"stream"},{"name":"stderr","text":"  5%|▍         | 41/844 [09:28<3:06:34, 13.94s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 36, 3]), contains NaN: False\n✅ TM-score: 0.3996\n\n▶ 41: 7MSF_S | len=14 | TM=None\n✅ Loaded MSA for 7MSF_S from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 7MSF_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▍         | 42/844 [09:42<3:05:46, 13.90s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.1703\n\n▶ 42: 5MSF_S | len=18 | TM=None\n✅ Loaded MSA for 5MSF_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 18])\n⚠️ Skipped msa_feat for 5MSF_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▌         | 43/844 [09:55<3:05:28, 13.89s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 18, 3]), contains NaN: False\n✅ TM-score: 0.2107\n\n▶ 43: 1LDZ_A | len=30 | TM=None\n✅ Loaded MSA for 1LDZ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([4, 30])\n⚠️ Skipped msa_feat for 1LDZ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▌         | 44/844 [10:10<3:06:18, 13.97s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.3122\n\n▶ 44: 1ZDK_S | len=19 | TM=None\n✅ Loaded MSA for 1ZDK_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([7, 19])\n⚠️ Skipped msa_feat for 1ZDK_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▌         | 45/844 [10:23<3:05:02, 13.90s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3184\n\n▶ 45: 1BVJ_A | len=23 | TM=None\n✅ Loaded MSA for 1BVJ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([3, 23])\n⚠️ Skipped msa_feat for 1BVJ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  5%|▌         | 46/844 [10:37<3:04:49, 13.90s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.2862\n\n▶ 46: 1B36_A | len=38 | TM=None\n✅ Loaded MSA for 1B36_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 38])\n✅ Injected msa_feat for 1B36_A | shape=torch.Size([64, 38])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 38])\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 47/844 [10:51<3:05:03, 13.93s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 38, 3]), contains NaN: False\n✅ TM-score: 0.2334\n\n▶ 47: 1HVU_I | len=30 | TM=None\n✅ Loaded MSA for 1HVU_I from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([13, 30])\n✅ Injected msa_feat for 1HVU_I | shape=torch.Size([13, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([13, 30])\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 48/844 [11:05<3:05:03, 13.95s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.1796\n\n▶ 48: 1BGZ_A | len=23 | TM=None\n✅ Loaded MSA for 1BGZ_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([8, 23])\n⚠️ Skipped msa_feat for 1BGZ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 49/844 [11:19<3:04:19, 13.91s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.2086\n\n▶ 49: 2BJ2_B | len=19 | TM=None\n✅ Loaded MSA for 2BJ2_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 19])\n⚠️ Skipped msa_feat for 2BJ2_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 50/844 [11:33<3:02:57, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.2465\n\n▶ 50: 2BJ2_A | len=21 | TM=None\n✅ Loaded MSA for 2BJ2_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 21])\n⚠️ Skipped msa_feat for 2BJ2_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 51/844 [11:46<3:02:25, 13.80s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.2456\n\n▶ 51: 28SP_A | len=28 | TM=None\n✅ Loaded MSA for 28SP_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 28])\n✅ Injected msa_feat for 28SP_A | shape=torch.Size([64, 28])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 28])\n","output_type":"stream"},{"name":"stderr","text":"  6%|▌         | 52/844 [12:00<3:02:11, 13.80s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 28, 3]), contains NaN: False\n✅ TM-score: 0.3780\n\n▶ 52: 1QFQ_A | len=15 | TM=None\n✅ Loaded MSA for 1QFQ_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 15])\n⚠️ Skipped msa_feat for 1QFQ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  6%|▋         | 53/844 [12:14<3:00:46, 13.71s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2396\n\n▶ 53: 17RA_A | len=21 | TM=None\n✅ Loaded MSA for 17RA_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([5, 21])\n⚠️ Skipped msa_feat for 17RA_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  6%|▋         | 54/844 [12:27<3:00:41, 13.72s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.1989\n\n▶ 54: 1BAU_B | len=23 | TM=None\n✅ Loaded MSA for 1BAU_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([7, 23])\n⚠️ Skipped msa_feat for 1BAU_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 55/844 [12:41<3:00:37, 13.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.1891\n\n▶ 55: 1BZ2_A | len=17 | TM=None\n✅ Loaded MSA for 1BZ2_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 17])\n✅ Injected msa_feat for 1BZ2_A | shape=torch.Size([64, 17])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 17])\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 56/844 [12:55<3:01:01, 13.78s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.3073\n\n▶ 56: 1CQ5_A | len=43 | TM=None\n✅ Loaded MSA for 1CQ5_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 43])\n✅ Injected msa_feat for 1CQ5_A | shape=torch.Size([64, 43])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 43])\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 57/844 [13:09<3:01:40, 13.85s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 43, 3]), contains NaN: False\n✅ TM-score: 0.3528\n\n▶ 57: 1QC8_A | len=25 | TM=None\n✅ Loaded MSA for 1QC8_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([5, 25])\n⚠️ Skipped msa_feat for 1QC8_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 58/844 [13:23<3:02:01, 13.89s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 25, 3]), contains NaN: False\n✅ TM-score: 0.2072\n\n▶ 58: 484D_B | len=27 | TM=None\n✅ Loaded MSA for 484D_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([9, 27])\n⚠️ Skipped msa_feat for 484D_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 59/844 [13:37<3:01:50, 13.90s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.3056\n\n▶ 59: 1D6K_B | len=37 | TM=None\n✅ Loaded MSA for 1D6K_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 37])\n✅ Injected msa_feat for 1D6K_B | shape=torch.Size([64, 37])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 37])\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 60/844 [13:51<3:01:57, 13.93s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 37, 3]), contains NaN: False\n✅ TM-score: 0.3343\n\n▶ 60: 1EIY_C | len=76 | TM=None\n✅ Loaded MSA for 1EIY_C from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 76])\n✅ Injected msa_feat for 1EIY_C | shape=torch.Size([64, 76])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 76])\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 61/844 [14:07<3:08:32, 14.45s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 76, 3]), contains NaN: False\n✅ TM-score: 0.4812\n\n▶ 61: 1ESH_A | len=13 | TM=None\n✅ Loaded MSA for 1ESH_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1ESH_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 62/844 [14:20<3:05:06, 14.20s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.1872\n\n▶ 62: 1EXY_A | len=33 | TM=None\n✅ Loaded MSA for 1EXY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([25, 33])\n✅ Injected msa_feat for 1EXY_A | shape=torch.Size([25, 33])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([25, 33])\n","output_type":"stream"},{"name":"stderr","text":"  7%|▋         | 63/844 [14:34<3:03:30, 14.10s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 33, 3]), contains NaN: False\n✅ TM-score: 0.1870\n\n▶ 63: 1D0U_A | len=21 | TM=None\n✅ Loaded MSA for 1D0U_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([2, 21])\n⚠️ Skipped msa_feat for 1D0U_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 64/844 [14:48<3:01:53, 13.99s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.2060\n\n▶ 64: 1ESY_A | len=19 | TM=None\n✅ Loaded MSA for 1ESY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([3, 19])\n⚠️ Skipped msa_feat for 1ESY_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 65/844 [15:02<3:00:35, 13.91s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.1540\n\n▶ 65: 1EUQ_B | len=72 | TM=None\n✅ Loaded MSA for 1EUQ_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 72])\n✅ Injected msa_feat for 1EUQ_B | shape=torch.Size([64, 72])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 72])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 66/844 [15:18<3:08:15, 14.52s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 72, 3]), contains NaN: False\n✅ TM-score: 0.6484\n\n▶ 66: 1F9L_A | len=22 | TM=None\n✅ Loaded MSA for 1F9L_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([39, 22])\n✅ Injected msa_feat for 1F9L_A | shape=torch.Size([39, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([39, 22])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 67/844 [15:31<3:05:07, 14.29s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.1665\n\n▶ 67: 1FFK_9 | len=122 | TM=None\n✅ Loaded MSA for 1FFK_9 from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 122])\n✅ Injected msa_feat for 1FFK_9 | shape=torch.Size([64, 122])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 122])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 68/844 [15:50<3:22:06, 15.63s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 122, 3]), contains NaN: False\n✅ TM-score: 0.6199\n\n▶ 68: 1EKZ_B | len=30 | TM=None\n✅ Loaded MSA for 1EKZ_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 30])\n✅ Injected msa_feat for 1EKZ_B | shape=torch.Size([64, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 30])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 69/844 [16:04<3:16:39, 15.23s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.1815\n\n▶ 69: 1F5U_B | len=18 | TM=None\n✅ Loaded MSA for 1F5U_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([15, 18])\n✅ Injected msa_feat for 1F5U_B | shape=torch.Size([15, 18])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([15, 18])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 70/844 [16:19<3:13:19, 14.99s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 18, 3]), contains NaN: False\n✅ TM-score: 0.3065\n\n▶ 70: 1F6Z_A | len=27 | TM=None\n✅ Loaded MSA for 1F6Z_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 27])\n✅ Injected msa_feat for 1F6Z_A | shape=torch.Size([64, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 27])\n","output_type":"stream"},{"name":"stderr","text":"  8%|▊         | 71/844 [16:33<3:09:55, 14.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.1703\n\n▶ 71: 1F6X_A | len=27 | TM=None\n✅ Loaded MSA for 1F6X_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([46, 27])\n✅ Injected msa_feat for 1F6X_A | shape=torch.Size([46, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([46, 27])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▊         | 72/844 [16:47<3:07:30, 14.57s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.2332\n\n▶ 72: 1F85_A | len=14 | TM=None\n✅ Loaded MSA for 1F85_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 1F85_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  9%|▊         | 73/844 [17:01<3:04:34, 14.36s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.2157\n\n▶ 73: 1F84_A | len=29 | TM=None\n✅ Loaded MSA for 1F84_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 1F84_A | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 74/844 [17:15<3:02:56, 14.26s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.2045\n\n▶ 74: 1FOQ_A | len=120 | TM=None\n✅ Loaded MSA for 1FOQ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([13, 120])\n✅ Injected msa_feat for 1FOQ_A | shape=torch.Size([13, 120])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([13, 120])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 75/844 [17:33<3:18:40, 15.50s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 120, 3]), contains NaN: False\n✅ TM-score: 0.3711\n\n▶ 75: 1FJE_A | len=22 | TM=None\n✅ Loaded MSA for 1FJE_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([4, 22])\n⚠️ Skipped msa_feat for 1FJE_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 76/844 [17:47<3:12:30, 15.04s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2174\n\n▶ 76: 1E4P_A | len=24 | TM=None\n✅ Loaded MSA for 1E4P_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([10, 24])\n✅ Injected msa_feat for 1E4P_A | shape=torch.Size([10, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 24])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 77/844 [18:02<3:09:01, 14.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.2698\n\n▶ 77: 1FQZ_A | len=27 | TM=None\n✅ Loaded MSA for 1FQZ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 27])\n✅ Injected msa_feat for 1FQZ_A | shape=torch.Size([64, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 27])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 78/844 [18:16<3:05:56, 14.56s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.4699\n\n▶ 78: 1G70_A | len=32 | TM=None\n✅ Loaded MSA for 1G70_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([10, 32])\n✅ Injected msa_feat for 1G70_A | shape=torch.Size([10, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 32])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 79/844 [18:29<3:03:04, 14.36s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.1974\n\n▶ 79: 1E7K_D | len=22 | TM=None\n✅ Loaded MSA for 1E7K_D from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 22])\n✅ Injected msa_feat for 1E7K_D | shape=torch.Size([64, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 22])\n","output_type":"stream"},{"name":"stderr","text":"  9%|▉         | 80/844 [18:43<3:00:48, 14.20s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2734\n\n▶ 80: 1FYO_A | len=27 | TM=None\n✅ Loaded MSA for 1FYO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([26, 27])\n✅ Injected msa_feat for 1FYO_A | shape=torch.Size([26, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([26, 27])\n","output_type":"stream"},{"name":"stderr","text":" 10%|▉         | 81/844 [18:57<2:59:23, 14.11s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.2625\n\n▶ 81: 1FHK_A | len=14 | TM=None\n✅ Loaded MSA for 1FHK_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 1FHK_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 10%|▉         | 82/844 [19:11<2:57:21, 13.97s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.1818\n\n▶ 82: 1I3X_A | len=19 | TM=None\n✅ Loaded MSA for 1I3X_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([3, 19])\n⚠️ Skipped msa_feat for 1I3X_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 10%|▉         | 83/844 [19:25<2:56:35, 13.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.2524\n\n▶ 83: 1I4C_A | len=13 | TM=None\n✅ Loaded MSA for 1I4C_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1I4C_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 10%|▉         | 84/844 [19:38<2:55:14, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2521\n\n▶ 84: 1I46_A | len=13 | TM=None\n✅ Loaded MSA for 1I46_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1I46_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 10%|█         | 85/844 [19:52<2:53:22, 13.71s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2442\n\n▶ 85: 1IBM_Y | len=15 | TM=None\n✅ Loaded MSA for 1IBM_Y from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([10, 15])\n✅ Injected msa_feat for 1IBM_Y | shape=torch.Size([10, 15])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 15])\n","output_type":"stream"},{"name":"stderr","text":" 10%|█         | 86/844 [20:05<2:51:45, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2721\n\n▶ 86: 1HWQ_A | len=30 | TM=None\n✅ Loaded MSA for 1HWQ_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([34, 30])\n✅ Injected msa_feat for 1HWQ_A | shape=torch.Size([34, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([34, 30])\n","output_type":"stream"},{"name":"stderr","text":" 10%|█         | 87/844 [20:19<2:52:42, 13.69s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.1473\n\n▶ 87: 1IK1_A | len=14 | TM=None\n✅ Loaded MSA for 1IK1_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 1IK1_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 10%|█         | 88/844 [20:32<2:51:31, 13.61s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.4280\n\n▶ 88: 1E95_A | len=36 | TM=None\n✅ Loaded MSA for 1E95_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 36])\n✅ Injected msa_feat for 1E95_A | shape=torch.Size([64, 36])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 36])\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 89/844 [20:46<2:53:08, 13.76s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 36, 3]), contains NaN: False\n✅ TM-score: 0.2508\n\n▶ 89: 1HS2_A | len=13 | TM=None\n✅ Loaded MSA for 1HS2_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1HS2_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 90/844 [21:00<2:51:40, 13.66s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.3443\n\n▶ 90: 1JO7_A | len=31 | TM=None\n✅ Loaded MSA for 1JO7_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([42, 31])\n✅ Injected msa_feat for 1JO7_A | shape=torch.Size([42, 31])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([42, 31])\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 91/844 [21:14<2:52:06, 13.71s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 31, 3]), contains NaN: False\n✅ TM-score: 0.2623\n\n▶ 91: 1IDV_A | len=10 | TM=None\n✅ Loaded MSA for 1IDV_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 10])\n⚠️ Skipped msa_feat for 1IDV_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 92/844 [21:27<2:50:24, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 10, 3]), contains NaN: False\n✅ TM-score: 0.1809\n\n▶ 92: 1K5I_A | len=23 | TM=None\n✅ Loaded MSA for 1K5I_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 23])\n✅ Injected msa_feat for 1K5I_A | shape=torch.Size([64, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 23])\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 93/844 [21:41<2:50:18, 13.61s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.4082\n\n▶ 93: 1K9W_A | len=23 | TM=None\n✅ Loaded MSA for 1K9W_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 23])\n✅ Injected msa_feat for 1K9W_A | shape=torch.Size([64, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 23])\n","output_type":"stream"},{"name":"stderr","text":" 11%|█         | 94/844 [21:55<2:50:50, 13.67s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.3119\n\n▶ 94: 1JTW_A | len=16 | TM=None\n✅ Loaded MSA for 1JTW_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([5, 16])\n⚠️ Skipped msa_feat for 1JTW_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 11%|█▏        | 95/844 [22:08<2:49:42, 13.59s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 16, 3]), contains NaN: False\n✅ TM-score: 0.5838\n\n▶ 95: 1K6G_A | len=22 | TM=None\n✅ Loaded MSA for 1K6G_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 22])\n⚠️ Skipped msa_feat for 1K6G_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 11%|█▏        | 96/844 [22:22<2:49:44, 13.62s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.4674\n\n▶ 96: 1K4B_A | len=14 | TM=None\n✅ Loaded MSA for 1K4B_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 1K4B_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 11%|█▏        | 97/844 [22:35<2:49:08, 13.59s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.2639\n\n▶ 97: 1K4A_A | len=14 | TM=None\n✅ Loaded MSA for 1K4A_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([3, 14])\n⚠️ Skipped msa_feat for 1K4A_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 98/844 [22:49<2:49:07, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.1613\n\n▶ 98: 1K6H_A | len=22 | TM=None\n✅ Loaded MSA for 1K6H_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([2, 22])\n⚠️ Skipped msa_feat for 1K6H_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 99/844 [23:02<2:48:52, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.3902\n\n▶ 99: 1JUR_A | len=22 | TM=None\n✅ Loaded MSA for 1JUR_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([12, 22])\n✅ Injected msa_feat for 1JUR_A | shape=torch.Size([12, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([12, 22])\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 100/844 [23:16<2:49:01, 13.63s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2659\n\n▶ 100: 1KKS_A | len=24 | TM=None\n✅ Loaded MSA for 1KKS_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 1KKS_A | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 101/844 [23:30<2:48:57, 13.64s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.3198\n\n▶ 101: 1JWC_A | len=28 | TM=None\n✅ Loaded MSA for 1JWC_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 28])\n✅ Injected msa_feat for 1JWC_A | shape=torch.Size([64, 28])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 28])\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 102/844 [23:43<2:49:10, 13.68s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 28, 3]), contains NaN: False\n✅ TM-score: 0.2286\n\n▶ 102: 1L1C_C | len=29 | TM=None\n✅ Loaded MSA for 1L1C_C from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 1L1C_C | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 103/844 [23:57<2:49:43, 13.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.2098\n\n▶ 103: 1KP7_A | len=30 | TM=None\n✅ Loaded MSA for 1KP7_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([2, 30])\n⚠️ Skipped msa_feat for 1KP7_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 104/844 [24:11<2:50:00, 13.78s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.2469\n\n▶ 104: 1JOX_A | len=21 | TM=None\n✅ Loaded MSA for 1JOX_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([8, 21])\n⚠️ Skipped msa_feat for 1JOX_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 12%|█▏        | 105/844 [24:25<2:50:06, 13.81s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.1615\n\n▶ 105: 1K2G_A | len=22 | TM=None\n✅ Loaded MSA for 1K2G_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([2, 22])\n⚠️ Skipped msa_feat for 1K2G_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 106/844 [24:39<2:49:53, 13.81s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2886\n\n▶ 106: 1L1W_A | len=29 | TM=None\n✅ Loaded MSA for 1L1W_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 1L1W_A | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 107/844 [24:53<2:49:23, 13.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.3451\n\n▶ 107: 1LC6_A | len=24 | TM=None\n✅ Loaded MSA for 1LC6_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 1LC6_A | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 108/844 [25:06<2:49:04, 13.78s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.1963\n\n▶ 108: 1LS2_B | len=76 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 109: 1KKA_A | len=17 | TM=None\n✅ Loaded MSA for 1KKA_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([32, 17])\n✅ Injected msa_feat for 1KKA_A | shape=torch.Size([32, 17])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([32, 17])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 110/844 [25:20<2:09:14, 10.56s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.0979\n\n▶ 110: 1MFY_A | len=31 | TM=None\n✅ Loaded MSA for 1MFY_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([47, 31])\n✅ Injected msa_feat for 1MFY_A | shape=torch.Size([47, 31])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([47, 31])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 111/844 [25:34<2:19:12, 11.40s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 31, 3]), contains NaN: False\n✅ TM-score: 0.1891\n\n▶ 111: 1M5L_A | len=38 | TM=None\n✅ Loaded MSA for 1M5L_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([62, 38])\n✅ Injected msa_feat for 1M5L_A | shape=torch.Size([62, 38])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([62, 38])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 112/844 [25:48<2:26:36, 12.02s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 38, 3]), contains NaN: False\n✅ TM-score: 0.2193\n\n▶ 112: 1MFJ_A | len=20 | TM=None\n✅ Loaded MSA for 1MFJ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([35, 20])\n✅ Injected msa_feat for 1MFJ_A | shape=torch.Size([35, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([35, 20])\n","output_type":"stream"},{"name":"stderr","text":" 13%|█▎        | 113/844 [26:02<2:32:21, 12.51s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.4055\n\n▶ 113: 1MFK_A | len=23 | TM=None\n✅ Loaded MSA for 1MFK_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([18, 23])\n✅ Injected msa_feat for 1MFK_A | shape=torch.Size([18, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([18, 23])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▎        | 114/844 [26:15<2:36:06, 12.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.3154\n\n▶ 114: 1N34_A | len=1513 | TM=None\n⏭️ Sequence too long → skip\n\n▶ 115: 1JTJ_A | len=23 | TM=None\n✅ Loaded MSA for 1JTJ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 23])\n✅ Injected msa_feat for 1JTJ_A | shape=torch.Size([64, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 23])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▎        | 116/844 [26:29<2:03:29, 10.18s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.1596\n\n▶ 116: 1MT4_A | len=24 | TM=None\n✅ Loaded MSA for 1MT4_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 1MT4_A | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 117/844 [26:43<2:14:03, 11.06s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.4414\n\n▶ 117: 1MNX_A | len=42 | TM=None\n✅ Loaded MSA for 1MNX_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 42])\n✅ Injected msa_feat for 1MNX_A | shape=torch.Size([64, 42])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 42])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 118/844 [26:57<2:22:40, 11.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 42, 3]), contains NaN: False\n✅ TM-score: 0.4102\n\n▶ 118: 1NA2_A | len=30 | TM=None\n✅ Loaded MSA for 1NA2_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 30])\n✅ Injected msa_feat for 1NA2_A | shape=torch.Size([64, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 30])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 119/844 [27:11<2:29:38, 12.38s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.2176\n\n▶ 119: 1NC0_A | len=24 | TM=None\n✅ Loaded MSA for 1NC0_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([4, 24])\n⚠️ Skipped msa_feat for 1NC0_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 120/844 [27:25<2:34:14, 12.78s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.2009\n\n▶ 120: 1N8X_A | len=36 | TM=None\n✅ Loaded MSA for 1N8X_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 36])\n✅ Injected msa_feat for 1N8X_A | shape=torch.Size([64, 36])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 36])\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 121/844 [27:39<2:38:15, 13.13s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 36, 3]), contains NaN: False\n✅ TM-score: 0.3377\n\n▶ 121: 1OQ0_A | len=15 | TM=None\n✅ Loaded MSA for 1OQ0_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 15])\n⚠️ Skipped msa_feat for 1OQ0_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 14%|█▍        | 122/844 [27:52<2:39:23, 13.25s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.1522\n\n▶ 122: 1OW9_A | len=23 | TM=None\n✅ Loaded MSA for 1OW9_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([13, 23])\n✅ Injected msa_feat for 1OW9_A | shape=torch.Size([13, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([13, 23])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▍        | 123/844 [28:06<2:41:05, 13.41s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.2698\n\n▶ 123: 1OSW_A | len=23 | TM=None\n✅ Loaded MSA for 1OSW_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 23])\n⚠️ Skipped msa_feat for 1OSW_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▍        | 124/844 [28:20<2:41:42, 13.48s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.2090\n\n▶ 124: 1M82_A | len=25 | TM=None\n✅ Loaded MSA for 1M82_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([59, 25])\n✅ Injected msa_feat for 1M82_A | shape=torch.Size([59, 25])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([59, 25])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▍        | 125/844 [28:33<2:42:54, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 25, 3]), contains NaN: False\n✅ TM-score: 0.2554\n\n▶ 125: 1NYB_B | len=24 | TM=None\n✅ Loaded MSA for 1NYB_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([10, 24])\n✅ Injected msa_feat for 1NYB_B | shape=torch.Size([10, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 24])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▍        | 126/844 [28:47<2:43:09, 13.64s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.2604\n\n▶ 126: 1PJY_A | len=22 | TM=None\n✅ Loaded MSA for 1PJY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 22])\n✅ Injected msa_feat for 1PJY_A | shape=torch.Size([64, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 22])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▌        | 127/844 [29:01<2:43:06, 13.65s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2256\n\n▶ 127: 1P6V_B | len=68 | TM=None\n✅ Loaded MSA for 1P6V_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 68])\n✅ Injected msa_feat for 1P6V_B | shape=torch.Size([64, 68])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 68])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▌        | 128/844 [29:16<2:48:48, 14.15s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 68, 3]), contains NaN: False\n✅ TM-score: 0.3676\n\n▶ 128: 1P6V_D | len=68 | TM=None\n✅ Loaded MSA for 1P6V_D from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 68])\n✅ Injected msa_feat for 1P6V_D | shape=torch.Size([64, 68])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 68])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▌        | 129/844 [29:31<2:52:33, 14.48s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 68, 3]), contains NaN: False\n✅ TM-score: 0.4448\n\n▶ 129: 1N66_A | len=22 | TM=None\n✅ Loaded MSA for 1N66_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([11, 22])\n✅ Injected msa_feat for 1N66_A | shape=torch.Size([11, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 22])\n","output_type":"stream"},{"name":"stderr","text":" 15%|█▌        | 130/844 [29:45<2:50:07, 14.30s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2228\n\n▶ 130: 1HS1_A | len=13 | TM=None\n✅ Loaded MSA for 1HS1_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1HS1_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 131/844 [29:59<2:47:22, 14.09s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.3144\n\n▶ 131: 1HS8_A | len=13 | TM=None\n✅ Loaded MSA for 1HS8_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1HS8_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 132/844 [30:12<2:45:19, 13.93s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2667\n\n▶ 132: 1HS4_A | len=13 | TM=None\n✅ Loaded MSA for 1HS4_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1HS4_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 133/844 [30:26<2:43:54, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2868\n\n▶ 133: 1HS3_A | len=13 | TM=None\n✅ Loaded MSA for 1HS3_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1HS3_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 134/844 [30:40<2:42:36, 13.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2966\n\n▶ 134: 1JZC_A | len=13 | TM=None\n✅ Loaded MSA for 1JZC_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 13])\n⚠️ Skipped msa_feat for 1JZC_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 135/844 [30:53<2:41:46, 13.69s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 13, 3]), contains NaN: False\n✅ TM-score: 0.2264\n\n▶ 135: 1QZC_C | len=27 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 136: 1P5M_A | len=55 | TM=None\n✅ Loaded MSA for 1P5M_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 55])\n✅ Injected msa_feat for 1P5M_A | shape=torch.Size([64, 55])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 55])\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▌        | 137/844 [31:07<2:05:25, 10.64s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 55, 3]), contains NaN: False\n✅ TM-score: 0.3935\n\n▶ 137: 1R2W_C | len=58 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 138: 1P5N_A | len=34 | TM=None\n✅ Loaded MSA for 1P5N_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1P5N_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 16%|█▋        | 139/844 [31:21<1:48:02,  9.19s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.3371\n\n▶ 139: 1QZC_B | len=21 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 140: 1P5P_A | len=77 | TM=None\n✅ Loaded MSA for 1P5P_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 77])\n✅ Injected msa_feat for 1P5P_A | shape=torch.Size([64, 77])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 77])\n","output_type":"stream"},{"name":"stderr","text":" 17%|█▋        | 141/844 [31:37<1:42:06,  8.72s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 77, 3]), contains NaN: False\n✅ TM-score: 0.2596\n\n▶ 141: 1QZA_B | len=75 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 142: 1QZB_B | len=76 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 143: 1QWB_A | len=26 | TM=None\n✅ Loaded MSA for 1QWB_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([6, 26])\n⚠️ Skipped msa_feat for 1QWB_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 17%|█▋        | 144/844 [31:51<1:21:10,  6.96s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 26, 3]), contains NaN: False\n✅ TM-score: 0.1964\n\n▶ 144: 1QWA_A | len=21 | TM=None\n✅ Loaded MSA for 1QWA_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([9, 21])\n⚠️ Skipped msa_feat for 1QWA_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 17%|█▋        | 145/844 [32:05<1:34:32,  8.11s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.1892\n\n▶ 145: 1Q75_A | len=15 | TM=None\n✅ Loaded MSA for 1Q75_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 15])\n⚠️ Skipped msa_feat for 1Q75_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 17%|█▋        | 146/844 [32:18<1:46:48,  9.18s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2159\n\n▶ 146: 1R2P_A | len=34 | TM=None\n✅ Loaded MSA for 1R2P_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1R2P_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 17%|█▋        | 147/844 [32:32<1:58:37, 10.21s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2094\n\n▶ 147: 1RFR_A | len=30 | TM=None\n✅ Loaded MSA for 1RFR_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 30])\n✅ Injected msa_feat for 1RFR_A | shape=torch.Size([64, 30])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 30])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 148/844 [32:46<2:08:32, 11.08s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 30, 3]), contains NaN: False\n✅ TM-score: 0.2706\n\n▶ 148: 1S9S_A | len=101 | TM=None\n✅ Loaded MSA for 1S9S_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 101])\n✅ Injected msa_feat for 1S9S_A | shape=torch.Size([64, 101])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 101])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 149/844 [33:03<2:24:54, 12.51s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 101, 3]), contains NaN: False\n✅ TM-score: 0.3427\n\n▶ 149: 1RY1_E | len=50 | TM=None\n✅ Loaded MSA for 1RY1_E from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 50])\n✅ Injected msa_feat for 1RY1_E | shape=torch.Size([64, 50])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 50])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 150/844 [33:17<2:30:25, 13.01s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 50, 3]), contains NaN: False\n✅ TM-score: 0.4353\n\n▶ 150: 1SZY_A | len=21 | TM=None\n✅ Loaded MSA for 1SZY_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 21])\n✅ Injected msa_feat for 1SZY_A | shape=torch.Size([64, 21])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 21])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 151/844 [33:31<2:32:34, 13.21s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.4096\n\n▶ 151: 1ROQ_A | len=14 | TM=None\n✅ Loaded MSA for 1ROQ_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 1ROQ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 152/844 [33:44<2:33:23, 13.30s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.3590\n\n▶ 152: 1R7W_A | len=34 | TM=None\n✅ Loaded MSA for 1R7W_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1R7W_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 153/844 [33:58<2:35:01, 13.46s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.3091\n\n▶ 153: 1R7Z_A | len=34 | TM=None\n✅ Loaded MSA for 1R7Z_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1R7Z_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 154/844 [34:12<2:36:02, 13.57s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2471\n\n▶ 154: 1T4L_A | len=32 | TM=None\n✅ Loaded MSA for 1T4L_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 32])\n✅ Injected msa_feat for 1T4L_A | shape=torch.Size([64, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 32])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 155/844 [34:26<2:36:07, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.2711\n\n▶ 155: 1WKS_A | len=17 | TM=None\n✅ Loaded MSA for 1WKS_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([50, 17])\n✅ Injected msa_feat for 1WKS_A | shape=torch.Size([50, 17])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([50, 17])\n","output_type":"stream"},{"name":"stderr","text":" 18%|█▊        | 156/844 [34:39<2:36:31, 13.65s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.3646\n\n▶ 156: 1T28_A | len=34 | TM=None\n✅ Loaded MSA for 1T28_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1T28_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▊        | 157/844 [34:53<2:36:58, 13.71s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2092\n\n▶ 157: 1S34_A | len=23 | TM=None\n✅ Loaded MSA for 1S34_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([34, 23])\n✅ Injected msa_feat for 1S34_A | shape=torch.Size([34, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([34, 23])\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▊        | 158/844 [35:07<2:37:04, 13.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.4232\n\n▶ 158: 1TJZ_A | len=22 | TM=None\n✅ Loaded MSA for 1TJZ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([3, 22])\n⚠️ Skipped msa_feat for 1TJZ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▉        | 159/844 [35:21<2:37:10, 13.77s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.1828\n\n▶ 159: 1TXS_A | len=38 | TM=None\n✅ Loaded MSA for 1TXS_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 38])\n✅ Injected msa_feat for 1TXS_A | shape=torch.Size([64, 38])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 38])\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▉        | 160/844 [35:35<2:37:40, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 38, 3]), contains NaN: False\n✅ TM-score: 0.2808\n\n▶ 160: 1R4H_A | len=10 | TM=None\n✅ Loaded MSA for 1R4H_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 10])\n⚠️ Skipped msa_feat for 1R4H_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▉        | 161/844 [35:48<2:35:56, 13.70s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 10, 3]), contains NaN: False\n✅ TM-score: 0.1701\n\n▶ 161: 1XHP_A | len=32 | TM=None\n✅ Loaded MSA for 1XHP_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 32])\n✅ Injected msa_feat for 1XHP_A | shape=torch.Size([64, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 32])\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▉        | 162/844 [36:02<2:35:33, 13.69s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.3274\n\n▶ 162: 1XSG_A | len=27 | TM=None\n✅ Loaded MSA for 1XSG_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 27])\n✅ Injected msa_feat for 1XSG_A | shape=torch.Size([64, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 27])\n","output_type":"stream"},{"name":"stderr","text":" 19%|█▉        | 163/844 [36:16<2:35:40, 13.72s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.1757\n\n▶ 163: 1Y1Y_P | len=4 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 164: 1XSH_A | len=27 | TM=None\n✅ Loaded MSA for 1XSH_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 27])\n✅ Injected msa_feat for 1XSH_A | shape=torch.Size([64, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 27])\n","output_type":"stream"},{"name":"stderr","text":" 20%|█▉        | 165/844 [36:30<1:59:40, 10.58s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.2020\n\n▶ 165: 1YLG_A | len=31 | TM=None\n✅ Loaded MSA for 1YLG_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([47, 31])\n✅ Injected msa_feat for 1YLG_A | shape=torch.Size([47, 31])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([47, 31])\n","output_type":"stream"},{"name":"stderr","text":" 20%|█▉        | 166/844 [36:44<2:08:50, 11.40s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 31, 3]), contains NaN: False\n✅ TM-score: 0.1851\n\n▶ 166: 1YNE_A | len=31 | TM=None\n✅ Loaded MSA for 1YNE_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 31])\n✅ Injected msa_feat for 1YNE_A | shape=torch.Size([64, 31])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 31])\n","output_type":"stream"},{"name":"stderr","text":" 20%|█▉        | 167/844 [36:57<2:15:32, 12.01s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 31, 3]), contains NaN: False\n✅ TM-score: 0.2952\n\n▶ 167: 1XWU_A | len=16 | TM=None\n✅ Loaded MSA for 1XWU_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 16])\n⚠️ Skipped msa_feat for 1XWU_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 20%|█▉        | 168/844 [37:11<2:19:38, 12.39s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 16, 3]), contains NaN: False\n✅ TM-score: 0.3804\n\n▶ 168: 1XWP_A | len=15 | TM=None\n✅ Loaded MSA for 1XWP_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 15])\n⚠️ Skipped msa_feat for 1XWP_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 20%|██        | 169/844 [37:24<2:22:43, 12.69s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 15, 3]), contains NaN: False\n✅ TM-score: 0.2389\n\n▶ 169: 1TBK_A | len=17 | TM=None\n✅ Loaded MSA for 1TBK_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 17])\n⚠️ Skipped msa_feat for 1TBK_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 20%|██        | 170/844 [37:38<2:25:43, 12.97s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.2032\n\n▶ 170: 1YMO_A | len=47 | TM=None\n✅ Loaded MSA for 1YMO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 47])\n✅ Injected msa_feat for 1YMO_A | shape=torch.Size([64, 47])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 47])\n","output_type":"stream"},{"name":"stderr","text":" 20%|██        | 171/844 [37:52<2:28:28, 13.24s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 47, 3]), contains NaN: False\n✅ TM-score: 0.2599\n\n▶ 171: 1Z31_A | len=32 | TM=None\n✅ Loaded MSA for 1Z31_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([10, 32])\n✅ Injected msa_feat for 1Z31_A | shape=torch.Size([10, 32])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([10, 32])\n","output_type":"stream"},{"name":"stderr","text":" 20%|██        | 172/844 [38:05<2:29:23, 13.34s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 32, 3]), contains NaN: False\n✅ TM-score: 0.3305\n\n▶ 172: 1U63_D | len=49 | TM=None\n✅ Loaded MSA for 1U63_D from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([43, 49])\n✅ Injected msa_feat for 1U63_D | shape=torch.Size([43, 49])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([43, 49])\n","output_type":"stream"},{"name":"stderr","text":" 20%|██        | 173/844 [38:19<2:31:54, 13.58s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 49, 3]), contains NaN: False\n✅ TM-score: 0.5095\n\n▶ 173: 1ZC8_Z | len=91 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 174: 1ZC8_G | len=30 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 175: 1ZC8_J | len=52 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 176: 1ZC8_F | len=29 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 177: 1ZC8_I | len=49 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 178: 1ZC8_H | len=59 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 179: 1Z30_A | len=18 | TM=None\n✅ Loaded MSA for 1Z30_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 18])\n⚠️ Skipped msa_feat for 1Z30_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 21%|██▏       | 180/844 [38:33<54:14,  4.90s/it]  ","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 18, 3]), contains NaN: False\n✅ TM-score: 0.2472\n\n▶ 180: 1X18_D | len=24 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 181: 1X18_A | len=11 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 182: 1X18_B | len=31 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 183: 1ZC5_A | len=41 | TM=None\n✅ Loaded MSA for 1ZC5_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 41])\n✅ Injected msa_feat for 1ZC5_A | shape=torch.Size([64, 41])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 41])\n","output_type":"stream"},{"name":"stderr","text":" 22%|██▏       | 184/844 [38:47<48:01,  4.37s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 41, 3]), contains NaN: False\n✅ TM-score: 0.5270\n\n▶ 184: 1ZN1_B | len=59 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 185: 1Z2J_A | len=45 | TM=None\n✅ Loaded MSA for 1Z2J_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 45])\n✅ Injected msa_feat for 1Z2J_A | shape=torch.Size([64, 45])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 45])\n","output_type":"stream"},{"name":"stderr","text":" 22%|██▏       | 186/844 [39:01<53:40,  4.89s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 45, 3]), contains NaN: False\n✅ TM-score: 0.3723\n\n▶ 186: 1YSH_B | len=101 | TM=None\n✅ Loaded MSA for 1YSH_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 101])\n✅ Injected msa_feat for 1YSH_B | shape=torch.Size([64, 101])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 101])\n","output_type":"stream"},{"name":"stderr","text":" 22%|██▏       | 187/844 [39:18<1:10:48,  6.47s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 101, 3]), contains NaN: False\n✅ TM-score: 0.9034\n\n▶ 187: 1YSH_A | len=28 | TM=None\n✅ Loaded MSA for 1YSH_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 28])\n✅ Injected msa_feat for 1YSH_A | shape=torch.Size([64, 28])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 28])\n","output_type":"stream"},{"name":"stderr","text":" 22%|██▏       | 188/844 [39:31<1:23:13,  7.61s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 28, 3]), contains NaN: False\n✅ TM-score: 0.4668\n\n▶ 188: 1YSH_F | len=34 | TM=None\n✅ Loaded MSA for 1YSH_F from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 1YSH_F | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 22%|██▏       | 189/844 [39:45<1:35:35,  8.76s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.3533\n\n▶ 189: 1YSV_A | len=27 | TM=None\n✅ Loaded MSA for 1YSV_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([37, 27])\n✅ Injected msa_feat for 1YSV_A | shape=torch.Size([37, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([37, 27])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 190/844 [39:59<1:46:49,  9.80s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.3844\n\n▶ 190: 2ADT_A | len=43 | TM=None\n✅ Loaded MSA for 2ADT_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 43])\n✅ Injected msa_feat for 2ADT_A | shape=torch.Size([64, 43])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 43])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 191/844 [40:13<1:56:53, 10.74s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 43, 3]), contains NaN: False\n✅ TM-score: 0.3890\n\n▶ 191: 1WZ2_D | len=88 | TM=None\n✅ Loaded MSA for 1WZ2_D from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 88])\n✅ Injected msa_feat for 1WZ2_D | shape=torch.Size([64, 88])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 88])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 192/844 [40:29<2:11:11, 12.07s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 88, 3]), contains NaN: False\n✅ TM-score: 0.6768\n\n▶ 192: 2A64_A | len=417 | TM=None\n⏭️ Sequence too long → skip\n\n▶ 193: 2D19_A | len=17 | TM=None\n✅ Loaded MSA for 2D19_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([31, 17])\n✅ Injected msa_feat for 2D19_A | shape=torch.Size([31, 17])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([31, 17])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 194/844 [40:42<1:47:08,  9.89s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 17, 3]), contains NaN: False\n✅ TM-score: 0.3057\n\n▶ 194: 2D1B_B | len=39 | TM=None\n✅ Loaded MSA for 2D1B_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 39])\n✅ Injected msa_feat for 2D1B_B | shape=torch.Size([64, 39])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 39])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 195/844 [40:56<1:57:00, 10.82s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 39, 3]), contains NaN: False\n✅ TM-score: 0.3789\n\n▶ 195: 2EUY_A | len=34 | TM=None\n✅ Loaded MSA for 2EUY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 2EUY_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 196/844 [41:10<2:04:50, 11.56s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2615\n\n▶ 196: 2B6G_B | len=19 | TM=None\n✅ Loaded MSA for 2B6G_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([16, 19])\n✅ Injected msa_feat for 2B6G_B | shape=torch.Size([16, 19])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([16, 19])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 197/844 [41:24<2:10:39, 12.12s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3922\n\n▶ 197: 2ESE_B | len=23 | TM=None\n✅ Loaded MSA for 2ESE_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 23])\n✅ Injected msa_feat for 2ESE_B | shape=torch.Size([64, 23])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 23])\n","output_type":"stream"},{"name":"stderr","text":" 23%|██▎       | 198/844 [41:38<2:14:39, 12.51s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 23, 3]), contains NaN: False\n✅ TM-score: 0.3491\n\n▶ 198: 2F4X_B | len=24 | TM=None\n✅ Loaded MSA for 2F4X_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 2F4X_B | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▎       | 199/844 [41:51<2:17:58, 12.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.3235\n\n▶ 199: 2F88_A | len=34 | TM=None\n✅ Loaded MSA for 2F88_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 34])\n✅ Injected msa_feat for 2F88_A | shape=torch.Size([64, 34])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 34])\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▎       | 200/844 [42:05<2:20:41, 13.11s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 34, 3]), contains NaN: False\n✅ TM-score: 0.2617\n\n▶ 200: 2BQ5_S | len=19 | TM=None\n✅ Loaded MSA for 2BQ5_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 19])\n⚠️ Skipped msa_feat for 2BQ5_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 201/844 [42:19<2:21:55, 13.24s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.2783\n\n▶ 201: 2BS0_R | len=20 | TM=None\n✅ Loaded MSA for 2BS0_R from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([11, 20])\n✅ Injected msa_feat for 2BS0_R | shape=torch.Size([11, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 20])\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 202/844 [42:32<2:22:55, 13.36s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.4359\n\n▶ 202: 2BQ5_R | len=19 | TM=None\n✅ Loaded MSA for 2BQ5_R from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([6, 19])\n⚠️ Skipped msa_feat for 2BQ5_R | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 203/844 [42:46<2:23:05, 13.39s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3755\n\n▶ 203: 2BS0_S | len=20 | TM=None\n✅ Loaded MSA for 2BS0_S from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([11, 20])\n✅ Injected msa_feat for 2BS0_S | shape=torch.Size([11, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 20])\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 204/844 [42:59<2:23:39, 13.47s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.1453\n\n▶ 204: 2G1W_A | len=22 | TM=None\n✅ Loaded MSA for 2G1W_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([11, 22])\n✅ Injected msa_feat for 2G1W_A | shape=torch.Size([11, 22])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([11, 22])\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 205/844 [43:13<2:23:49, 13.51s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.1284\n\n▶ 205: 2B2E_R | len=19 | TM=None\n✅ Loaded MSA for 2B2E_R from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([6, 19])\n⚠️ Skipped msa_feat for 2B2E_R | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 24%|██▍       | 206/844 [43:27<2:23:55, 13.54s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3524\n\n▶ 206: 2B2E_S | len=19 | TM=None\n✅ Loaded MSA for 2B2E_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 19])\n⚠️ Skipped msa_feat for 2B2E_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▍       | 207/844 [43:40<2:23:48, 13.55s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.1904\n\n▶ 207: 2GO5_9 | len=90 | TM=None\n✅ Loaded MSA for 2GO5_9 from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 90])\n✅ Injected msa_feat for 2GO5_9 | shape=torch.Size([64, 90])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 90])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▍       | 208/844 [43:56<2:31:04, 14.25s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 90, 3]), contains NaN: False\n✅ TM-score: 0.6268\n\n▶ 208: 2GO5_A | len=127 | TM=None\n✅ Loaded MSA for 2GO5_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 127])\n✅ Injected msa_feat for 2GO5_A | shape=torch.Size([64, 127])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 127])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▍       | 209/844 [44:16<2:47:52, 15.86s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 127, 3]), contains NaN: False\n✅ TM-score: 0.6615\n\n▶ 209: 2GIP_A | len=28 | TM=None\n✅ Loaded MSA for 2GIP_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 28])\n✅ Injected msa_feat for 2GIP_A | shape=torch.Size([64, 28])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 28])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▍       | 210/844 [44:29<2:41:08, 15.25s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 28, 3]), contains NaN: False\n✅ TM-score: 0.3146\n\n▶ 210: 2GIO_A | len=29 | TM=None\n✅ Loaded MSA for 2GIO_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 29])\n✅ Injected msa_feat for 2GIO_A | shape=torch.Size([64, 29])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 29])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▌       | 211/844 [44:43<2:36:05, 14.80s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 29, 3]), contains NaN: False\n✅ TM-score: 0.3539\n\n▶ 211: 2FEY_A | len=43 | TM=None\n✅ Loaded MSA for 2FEY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([28, 43])\n✅ Injected msa_feat for 2FEY_A | shape=torch.Size([28, 43])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([28, 43])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▌       | 212/844 [44:57<2:32:42, 14.50s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 43, 3]), contains NaN: False\n✅ TM-score: 0.2246\n\n▶ 212: 2AKE_B | len=72 | TM=None\n✅ Loaded MSA for 2AKE_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 72])\n✅ Injected msa_feat for 2AKE_B | shape=torch.Size([64, 72])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 72])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▌       | 213/844 [45:13<2:36:21, 14.87s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 72, 3]), contains NaN: False\n✅ TM-score: 0.6588\n\n▶ 213: 2DR2_B | len=75 | TM=None\n✅ Loaded MSA for 2DR2_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 75])\n✅ Injected msa_feat for 2DR2_B | shape=torch.Size([64, 75])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 75])\n","output_type":"stream"},{"name":"stderr","text":" 25%|██▌       | 214/844 [45:28<2:37:59, 15.05s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 75, 3]), contains NaN: False\n✅ TM-score: 0.6036\n\n▶ 214: 2AGN_C | len=77 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 215: 2AGN_A | len=29 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 216: 2IZN_S | len=19 | TM=None\n✅ Loaded MSA for 2IZN_S from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 19])\n⚠️ Skipped msa_feat for 2IZN_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▌       | 217/844 [45:42<1:35:36,  9.15s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.1950\n\n▶ 217: 2IZ8_R | len=19 | TM=None\n✅ Loaded MSA for 2IZ8_R from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([8, 19])\n⚠️ Skipped msa_feat for 2IZ8_R | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▌       | 218/844 [45:55<1:45:18, 10.09s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3855\n\n▶ 218: 2IZ8_S | len=19 | TM=None\n✅ Loaded MSA for 2IZ8_S from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([8, 19])\n⚠️ Skipped msa_feat for 2IZ8_S | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▌       | 219/844 [46:09<1:53:46, 10.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 19, 3]), contains NaN: False\n✅ TM-score: 0.3275\n\n▶ 219: 2HGH_B | len=55 | TM=None\n✅ Loaded MSA for 2HGH_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 55])\n✅ Injected msa_feat for 2HGH_B | shape=torch.Size([64, 55])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 55])\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▌       | 220/844 [46:23<2:01:26, 11.68s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 55, 3]), contains NaN: False\n✅ TM-score: 0.4310\n\n▶ 220: 2F87_A | len=12 | TM=None\n✅ Loaded MSA for 2F87_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 12])\n⚠️ Skipped msa_feat for 2F87_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▌       | 221/844 [46:36<2:05:53, 12.12s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 12, 3]), contains NaN: False\n✅ TM-score: 0.1485\n\n▶ 221: 2GV3_A | len=22 | TM=None\n✅ Loaded MSA for 2GV3_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([3, 22])\n⚠️ Skipped msa_feat for 2GV3_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▋       | 222/844 [46:50<2:10:06, 12.55s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2133\n\n▶ 222: 2DER_D | len=76 | TM=None\n✅ Loaded MSA for 2DER_D from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 76])\n✅ Injected msa_feat for 2DER_D | shape=torch.Size([64, 76])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 76])\n","output_type":"stream"},{"name":"stderr","text":" 26%|██▋       | 223/844 [47:06<2:18:50, 13.42s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 76, 3]), contains NaN: False\n✅ TM-score: 0.6408\n\n▶ 223: 2DET_C | len=76 | TM=None\n✅ Loaded MSA for 2DET_C from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 76])\n✅ Injected msa_feat for 2DET_C | shape=torch.Size([64, 76])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 76])\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 224/844 [47:21<2:24:48, 14.01s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 76, 3]), contains NaN: False\n✅ TM-score: 0.6873\n\n▶ 224: 2EVY_A | len=14 | TM=None\n✅ Loaded MSA for 2EVY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([1, 14])\n⚠️ Skipped msa_feat for 2EVY_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 225/844 [47:35<2:22:39, 13.83s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 14, 3]), contains NaN: False\n✅ TM-score: 0.2831\n\n▶ 225: 2IXY_A | len=27 | TM=None\n✅ Loaded MSA for 2IXY_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 27])\n✅ Injected msa_feat for 2IXY_A | shape=torch.Size([64, 27])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 27])\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 226/844 [47:48<2:22:04, 13.79s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.2128\n\n▶ 226: 2IXZ_A | len=8 | TM=None\n✅ Loaded MSA for 2IXZ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([1, 8])\n⚠️ Skipped msa_feat for 2IXZ_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 227/844 [48:02<2:19:56, 13.61s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 8, 3]), contains NaN: False\n✅ TM-score: 0.0998\n\n▶ 227: 2HEM_A | len=24 | TM=None\n✅ Loaded MSA for 2HEM_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 24])\n✅ Injected msa_feat for 2HEM_A | shape=torch.Size([64, 24])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 24])\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 228/844 [48:15<2:20:11, 13.65s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 24, 3]), contains NaN: False\n✅ TM-score: 0.1879\n\n▶ 228: 1ZBH_E | len=20 | TM=None\n✅ Loaded MSA for 1ZBH_E from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 20])\n✅ Injected msa_feat for 1ZBH_E | shape=torch.Size([64, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 20])\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 229/844 [48:29<2:19:26, 13.60s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.3333\n\n▶ 229: 2AHT_A | len=27 | TM=None\n✅ Loaded MSA for 2AHT_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 27])\n⚠️ Skipped msa_feat for 2AHT_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 230/844 [48:43<2:19:38, 13.65s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 27, 3]), contains NaN: False\n✅ TM-score: 0.2612\n\n▶ 230: 2HNS_A | len=22 | TM=None\n✅ Loaded MSA for 2HNS_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([4, 22])\n⚠️ Skipped msa_feat for 2HNS_A | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 27%|██▋       | 231/844 [48:56<2:19:38, 13.67s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 22, 3]), contains NaN: False\n✅ TM-score: 0.2268\n\n▶ 231: 2IY3_B | len=110 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 232: 2J28_8 | len=74 | TM=None\n✅ Loaded MSA for 2J28_8 from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 74])\n✅ Injected msa_feat for 2J28_8 | shape=torch.Size([64, 74])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 74])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 233/844 [49:12<1:51:12, 10.92s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 74, 3]), contains NaN: False\n✅ TM-score: 0.4628\n\n▶ 233: 2J37_A | len=128 | TM=None\n✅ Loaded MSA for 2J37_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 128])\n✅ Injected msa_feat for 2J37_A | shape=torch.Size([64, 128])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 128])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 234/844 [49:32<2:13:26, 13.13s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 128, 3]), contains NaN: False\n✅ TM-score: 0.7864\n\n▶ 234: 2J28_A | len=117 | TM=None\n✅ Loaded MSA for 2J28_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 117])\n✅ Injected msa_feat for 2J28_A | shape=torch.Size([64, 117])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 117])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 235/844 [49:50<2:26:28, 14.43s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 117, 3]), contains NaN: False\n✅ TM-score: 0.7882\n\n▶ 235: 2NOQ_B | len=46 | TM=None\n✅ Loaded MSA for 2NOQ_B from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 46])\n✅ Injected msa_feat for 2NOQ_B | shape=torch.Size([64, 46])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 46])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 236/844 [50:03<2:24:27, 14.26s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 46, 3]), contains NaN: False\n✅ TM-score: 0.9154\n\n▶ 236: 2NOQ_E | len=53 | TM=None\n✅ Loaded MSA for 2NOQ_E from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 53])\n✅ Injected msa_feat for 2NOQ_E | shape=torch.Size([64, 53])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 53])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 237/844 [50:17<2:23:30, 14.19s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 53, 3]), contains NaN: False\n✅ TM-score: 0.6593\n\n▶ 237: 2NOQ_A | len=190 | TM=None\n✅ Loaded MSA for 2NOQ_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 190])\n✅ Injected msa_feat for 2NOQ_A | shape=torch.Size([64, 190])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 190])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 238/844 [50:51<3:19:57, 19.80s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 190, 3]), contains NaN: False\n✅ TM-score: 0.6243\n\n▶ 238: 2FDT_A | len=36 | TM=None\n✅ Loaded MSA for 2FDT_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 36])\n✅ Injected msa_feat for 2FDT_A | shape=torch.Size([64, 36])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 36])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 239/844 [51:05<3:02:06, 18.06s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 36, 3]), contains NaN: False\n✅ TM-score: 0.4782\n\n▶ 239: 2IHX_B | len=75 | TM=None\n✅ Loaded MSA for 2IHX_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([64, 75])\n✅ Injected msa_feat for 2IHX_B | shape=torch.Size([64, 75])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 75])\n","output_type":"stream"},{"name":"stderr","text":" 28%|██▊       | 240/844 [51:21<2:54:17, 17.31s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 75, 3]), contains NaN: False\n✅ TM-score: 0.2530\n\n▶ 240: 2OB7_D | len=86 | TM=None\n⏭️ Not enough ground-truth coordinates → skip\n\n▶ 241: 2FY1_B | len=21 | TM=None\n✅ Loaded MSA for 2FY1_B from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([6, 21])\n⚠️ Skipped msa_feat for 2FY1_B | depth insufficient or not found\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions'])\n❌ msa_feat not present\n","output_type":"stream"},{"name":"stderr","text":" 29%|██▊       | 242/844 [51:34<2:06:11, 12.58s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 21, 3]), contains NaN: False\n✅ TM-score: 0.3800\n\n▶ 242: 2O33_A | len=20 | TM=None\n✅ Loaded MSA for 2O33_A from /kaggle/input/stanford-rna-3d-folding/MSA_v2 | shape=torch.Size([26, 20])\n✅ Injected msa_feat for 2O33_A | shape=torch.Size([26, 20])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([26, 20])\n","output_type":"stream"},{"name":"stderr","text":" 29%|██▉       | 243/844 [51:48<2:08:50, 12.86s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 20, 3]), contains NaN: False\n✅ TM-score: 0.2889\n\n▶ 243: 2IL9_M | len=142 | TM=None\n✅ Loaded MSA for 2IL9_M from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 142])\n✅ Injected msa_feat for 2IL9_M | shape=torch.Size([64, 142])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 142])\n","output_type":"stream"},{"name":"stderr","text":" 29%|██▉       | 244/844 [52:11<2:35:31, 15.55s/it]","output_type":"stream"},{"name":"stdout","text":"🧪 Prediction shape: torch.Size([1, 142, 3]), contains NaN: False\n✅ TM-score: 0.3457\n\n▶ 244: 2IL9_A | len=142 | TM=None\n✅ Loaded MSA for 2IL9_A from /kaggle/input/stanford-rna-3d-folding/MSA | shape=torch.Size([64, 142])\n✅ Injected msa_feat for 2IL9_A | shape=torch.Size([64, 142])\n📦 input keys: dict_keys(['token_index', 'residue_index', 'asym_id', 'entity_id', 'sym_id', 'restype', 'token_bonds', 'ref_pos', 'ref_mask', 'ref_element', 'ref_charge', 'ref_atom_name_chars', 'ref_space_uid', 'has_frame', 'frame_atom_index', 'atom_to_token_idx', 'atom_to_tokatom_idx', 'is_protein', 'is_ligand', 'is_dna', 'is_rna', 'resolution', 'mol_id', 'mol_atom_index', 'entity_mol_id', 'pae_rep_atom_mask', 'plddt_m_rep_atom_mask', 'distogram_rep_atom_mask', 'modified_res_mask', 'bond_mask', 'msa', 'has_deletion', 'deletion_value', 'profile', 'deletion_mean', 'prot_pair_num_alignments', 'prot_unpair_num_alignments', 'rna_pair_num_alignments', 'rna_unpair_num_alignments', 'template_restype', 'template_all_atom_mask', 'template_all_atom_positions', 'msa_feat'])\n✅ msa_feat shape: torch.Size([64, 142])\n","output_type":"stream"}],"execution_count":null},{"cell_type":"code","source":"from scipy.ndimage import gaussian_filter1d\n\ndef smooth_coords(df, sigma=1.0):\n    df = df.copy()\n    for n in range(1, 6):  # 5개의 구조 (x_1 ~ x_5, y_1 ~ y_5, z_1 ~ z_5)\n        for axis in ['x', 'y', 'z']:\n            col = f\"{axis}_{n}\"\n            if col in df.columns:\n                df[col] = gaussian_filter1d(df[col].values, sigma=sigma)\n    return df","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"if MODEL_TYPE=='protenix' and not VALIDATION:\n    test_df=pd.read_csv('/kaggle/input/stanford-rna-3d-folding/test_sequences.csv')\n    import warnings\n    warnings.filterwarnings(\"ignore\")  \n    \n    dataset = DictDataset(test_df.sequence, dump_dir='output', id_list=test_df.target_id, use_msa_server=True)\n    num_data = len(dataset)\n    for i, seq in tqdm(enumerate(test_df.sequence),total=num_data):\n        try:\n            data, atom_array, data_error_message=dataset[i]\n            target_id = data[\"sample_name\"]\n            assert target_id==test_df.target_id[i]\n            assert data_error_message==''\n            \n            new_configs = update_inference_configs(configs, data[\"N_token\"].item())\n            runner.update_model_configs(new_configs)\n            prediction = runner.predict(data)\n            prediction=prediction['coordinate'][:,data['input_feature_dict']['atom_to_tokatom_idx']==12]\n\n            result = parse_output_to_df(prediction, seq, target_id)[0]\n        except:\n            target_id==test_df.target_id[i]\n            print('Failed to predict', target_id)\n            result=pd.DataFrame(columns=['ID', 'resname', 'resid', \n                                         'x_1', 'y_1', 'z_1', \n                                         'x_2', 'y_2', 'z_2',\n                                         'x_3', 'y_3', 'z_3', \n                                         'x_4', 'y_4', 'z_4', \n                                         'x_5', 'y_5', 'z_5'], \n                                         data=[[target_id, x, j+1] + [0.0]*15 for j, x in enumerate(seq)])\n        result = smooth_coords(result, sigma=1.0)\n        result['ID']=result.apply(lambda x: x.ID + '_' + str(x.resid), axis=1)\n        result.to_csv('submission.csv', index=False, mode='a', header=(i==0))\n        torch.cuda.empty_cache()\n\n    display(pd.read_csv('submission.csv'))","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"markdown","source":"","metadata":{}},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}